diff --git a/node_modules/@expo/xdl/build/ApiV2.js b/node_modules/@expo/xdl/build/ApiV2.js
index 08f20ee..cf00359 100644
--- a/node_modules/@expo/xdl/build/ApiV2.js
+++ b/node_modules/@expo/xdl/build/ApiV2.js
@@ -246,6 +246,13 @@ class ApiV2Client {
 
     try {
       response = await _axios().default.request(reqOptions);
+
+      if (process.env.CI) {
+        // Log the request options to help with debugging in case we have any issues here
+        const reqOptionsLog = {...reqOptions, data: { ...reqOptions.data, value: '(versions endpoint data)'}};
+        console.log(JSON.stringify(reqOptionsLog, null, 2));
+      }
+
       result = response.data;
     } catch (e) {
       var _e$response, _e$response$data, _e$response$data$erro;
