diff --git a/app/build.gradle b/app/build.gradle
index 2c1b374..6aefd77 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -49,8 +49,9 @@ dependencies {
     debugImplementation libs.androidx.ui.test.manifest
 }

-def projectRoot = rootDir.getAbsoluteFile().getParentFile().getAbsolutePath()
+def projectRoot = new File(rootDir.getAbsoluteFile(), 'my-project').getAbsolutePath()
 react {
+    root = new File(projectRoot)
     entryFile = file(["node", "-e", "require('expo/scripts/resolveAppEntry')", projectRoot, "android", "absolute"].execute(null, rootDir).text.trim())
     reactNativeDir = new File(["node", "--print", "require.resolve('react-native/package.json')"].execute(null, rootDir).text.trim()).getParentFile().getAbsoluteFile()
