diff --git a/node_modules/@expo/vector-icons/build/createIconSet.js b/node_modules/@expo/vector-icons/build/createIconSet.js
index 2d7e33f..cf513c8 100644
--- a/node_modules/@expo/vector-icons/build/createIconSet.js
+++ b/node_modules/@expo/vector-icons/build/createIconSet.js
@@ -30,7 +30,7 @@ export default function (glyphMap, fontName, expoAssetId, fontStyle) {
                 color: color,
                 size,
             });
-            return { uri: imagePath, scale: PixelRatio.get() };
+            return { ...imagePath, scale: PixelRatio.get() };
         };
         _mounted = false;
         _icon;
