diff --git a/dist/types/ReadonlyURL.d.ts b/dist/types/ReadonlyURL.d.ts
index 2fe7bbe..bb616e7 100644
--- a/dist/types/ReadonlyURL.d.ts
+++ b/dist/types/ReadonlyURL.d.ts
@@ -29,7 +29,7 @@ export interface ReadonlyURLSearchParams {
     thisArg: This,
   ): void;
   toString(): string;
-  @@iterator(): Iterator<[string, string]>;
+  [Symbol.iterator](): Iterator<[string, string]>;
 }
 /**
  * A readonly view of URL that prevents mutation of URL properties.
