diff --git a/node_modules/@shopify/react-native-skia/.DS_Store b/node_modules/@shopify/react-native-skia/.DS_Store
new file mode 100644
index 0000000..b7fa5d4
Binary files /dev/null and b/node_modules/@shopify/react-native-skia/.DS_Store differ
diff --git a/node_modules/@shopify/react-native-skia/android/.DS_Store b/node_modules/@shopify/react-native-skia/android/.DS_Store
new file mode 100644
index 0000000..8f70268
Binary files /dev/null and b/node_modules/@shopify/react-native-skia/android/.DS_Store differ
diff --git a/node_modules/@shopify/react-native-skia/android/CMakeLists.txt b/node_modules/@shopify/react-native-skia/android/CMakeLists.txt
index 5742abb..5047868 100644
--- a/node_modules/@shopify/react-native-skia/android/CMakeLists.txt
+++ b/node_modules/@shopify/react-native-skia/android/CMakeLists.txt
@@ -155,6 +155,7 @@ endif()
 message("-- JSI     : " ${JSI_LIB})
 
 unset(REACT_LIB CACHE)
+
 if(${REACT_NATIVE_VERSION} GREATER_EQUAL 71)
     # RN 0.71 distributes prebuilt binaries.
     set (REACT_LIB ReactAndroid::react_nativemodule_core)
@@ -227,15 +228,30 @@ message("-- TURBO   : " ${TURBOMODULES_LIB})
 add_definitions(-DREACT_NATIVE_VERSION=${REACT_NATIVE_VERSION})
 
 # Link
-target_link_libraries(
+
+
+if(${REACT_NATIVE_VERSION} GREATER_EQUAL 76)
+    target_link_libraries(
+        ${PACKAGE_NAME}
+        ReactAndroid::reactnative
+    )
+else()
+    target_link_libraries(
         ${PACKAGE_NAME}
-        ${LOG_LIB}
         ${FBJNI_LIBRARY}
         ${REACT_LIB}
         ${JSI_LIB}
         ${REACTNATIVEJNI_LIB}
         ${RUNTIMEEXECUTOR_LIB}
         ${TURBOMODULES_LIB}
+    )
+endif()
+
+target_link_libraries(
+        ${PACKAGE_NAME}
+        ${LOG_LIB}
+        ${FBJNI_LIBRARY}
+        ${JSI_LIB}
         ${SKIA_SVG_LIB}
         ${SKIA_SKSHAPER_LIB}
         ${SKIA_SKPARAGRAPH_LIB}
