diff --git a/node_modules/expo-router/src/static/renderStaticContent.tsx b/node_modules/expo-router/src/static/renderStaticContent.tsx
index 7fdae39..ec64189 100644
--- a/node_modules/expo-router/src/static/renderStaticContent.tsx
+++ b/node_modules/expo-router/src/static/renderStaticContent.tsx
@@ -6,6 +6,7 @@
  */
 
 import { ServerContainer, ServerContainerRef } from "@react-navigation/native";
+import * as Font from 'expo-font';
 import App, { getManifest } from "expo-router/_entry";
 import Head from "expo-router/head";
 import React from "react";
@@ -73,6 +74,9 @@ export function getStaticContent(location: URL): string {
 
   output = output.replace("</head>", `${css}</head>`);
 
+  // Inject static fonts loaded with expo-font
+  output = output.replace("</head>", `${Font._getStaticResources().join('')}</head>`);
+
   return "<!DOCTYPE html>" + output;
 }
 
