diff --git a/node_modules/lottie-react-native/android/src/main/java/com/airbnb/android/react/lottie/LottieAnimationViewPropertyManager.kt b/node_modules/lottie-react-native/android/src/main/java/com/airbnb/android/react/lottie/LottieAnimationViewPropertyManager.kt
index 2198eea..ae2c691 100644
--- a/node_modules/lottie-react-native/android/src/main/java/com/airbnb/android/react/lottie/LottieAnimationViewPropertyManager.kt
+++ b/node_modules/lottie-react-native/android/src/main/java/com/airbnb/android/react/lottie/LottieAnimationViewPropertyManager.kt
@@ -246,11 +246,11 @@ class LottieAnimationViewPropertyManager(view: LottieAnimationView) {
         colorFilter: ReadableMap,
         view: LottieAnimationView
     ) {
-        val color: Int = if (colorFilter.getType("color") == ReadableType.Map) {
+        val color: Int = (if (colorFilter.getType("color") == ReadableType.Map) {
             ColorPropConverter.getColor(colorFilter.getMap("color"), view.context)
         } else {
             colorFilter.getInt("color")
-        }
+        }) ?: 0

         val path = colorFilter.getString("keypath")
         val pathGlob = "$path.**"
