diff --git a/node_modules/lottie-react-native/ios/Fabric/LottieAnimationViewComponentView.mm b/node_modules/lottie-react-native/ios/Fabric/LottieAnimationViewComponentView.mm
index de8b545..085eebf 100644
--- a/node_modules/lottie-react-native/ios/Fabric/LottieAnimationViewComponentView.mm
+++ b/node_modules/lottie-react-native/ios/Fabric/LottieAnimationViewComponentView.mm
@@ -5,7 +5,7 @@
 #import <react/renderer/components/lottiereactnative/Props.h>
 #import <react/renderer/components/lottiereactnative/RCTComponentViewHelpers.h>
 
-#import "RCTFabricComponentsPlugins.h"
+#import <React/RCTFabricComponentsPlugins.h>
 #import <React/RCTView.h>
 #import "LottieContainerView.h"
 #import "RNLottieFabricConversions.h"
diff --git a/node_modules/lottie-react-native/ios/LottieReactNative/AnimationViewManagerModule.swift b/node_modules/lottie-react-native/ios/LottieReactNative/AnimationViewManagerModule.swift
index 83beee3..6474f53 100644
--- a/node_modules/lottie-react-native/ios/LottieReactNative/AnimationViewManagerModule.swift
+++ b/node_modules/lottie-react-native/ios/LottieReactNative/AnimationViewManagerModule.swift
@@ -3,22 +3,22 @@ import Lottie
 #if os(OSX)
 import AppKit
 
-typealias View = NSView
+public typealias View = NSView
 
 #else
 import UIKit
 
-typealias View = UIView
+public typealias View = UIView
 
 #endif
 
 @objc(LottieAnimationView)
-class AnimationViewManagerModule: RCTViewManager {
-    override func view() -> View! {
+open class AnimationViewManagerModule: RCTViewManager {
+    override public func view() -> View! {
         return ContainerView()
     }
 
-    @objc override func constantsToExport() -> [AnyHashable: Any]! {
+    @objc override open func constantsToExport() -> [AnyHashable: Any]! {
         return ["VERSION": 1]
     }
 
@@ -82,7 +82,7 @@ class AnimationViewManagerModule: RCTViewManager {
         }
     }
 
-    override static func requiresMainQueueSetup() -> Bool {
+    override public static func requiresMainQueueSetup() -> Bool {
         return true
     }
 }
diff --git a/node_modules/lottie-react-native/ios/LottieReactNative/ContainerView.swift b/node_modules/lottie-react-native/ios/LottieReactNative/ContainerView.swift
index 4c70405..fdc9a4d 100644
--- a/node_modules/lottie-react-native/ios/LottieReactNative/ContainerView.swift
+++ b/node_modules/lottie-react-native/ios/LottieReactNative/ContainerView.swift
@@ -1,5 +1,5 @@
-import Lottie
 import Foundation
+import Lottie
 
 @objc protocol LottieContainerViewDelegate {
     func onAnimationFinish(isCancelled: Bool)
@@ -12,7 +12,7 @@ import Foundation
  2- a LottieAnimationView that is a child of the RCTView and is bound to the same coordinates, just on top of it
  */
 @objc(LottieContainerView)
-class ContainerView: RCTView {
+open class ContainerView: RCTView {
     private var speed: CGFloat = 0.0
     private var progress: CGFloat = 0.0
     private var autoPlay: Bool = false
@@ -63,18 +63,20 @@ class ContainerView: RCTView {
         }
     }
 
-#if !(os(OSX))
-    override func traitCollectionDidChange(_ previousTraitCollection: UITraitCollection?) {
-        super.traitCollectionDidChange(previousTraitCollection)
-        if #available(iOS 13.0, tvOS 13.0, *) {
-            if self.traitCollection.hasDifferentColorAppearance(comparedTo: previousTraitCollection) {
-                if !colorFilters.isEmpty {
-                    applyColorProperties()
+    #if !(os(OSX))
+        override open func traitCollectionDidChange(_ previousTraitCollection: UITraitCollection?) {
+            super.traitCollectionDidChange(previousTraitCollection)
+            if #available(iOS 13.0, tvOS 13.0, *) {
+                if self.traitCollection.hasDifferentColorAppearance(
+                    comparedTo: previousTraitCollection)
+                {
+                    if !colorFilters.isEmpty {
+                        applyColorProperties()
+                    }
                 }
             }
         }
-    }
-#endif
+    #endif
 
     @objc func setSpeed(_ newSpeed: CGFloat) {
         speed = newSpeed
@@ -111,7 +113,8 @@ class ContainerView: RCTView {
             var filters = [String: String]()
             for filter in textFilters {
                 guard let key = filter.value(forKey: "keypath") as? String,
-                      let value = filter.value(forKey: "text") as? String else { break }
+                    let value = filter.value(forKey: "text") as? String
+                else { break }
                 filters[key] = value
             }
 
@@ -205,7 +208,8 @@ class ContainerView: RCTView {
         }
 
         guard let data = newSourceJson.data(using: String.Encoding.utf8),
-              let animation = try? JSONDecoder().decode(LottieAnimation.self, from: data) else {
+            let animation = try? JSONDecoder().decode(LottieAnimation.self, from: data)
+        else {
             failureCallback("Unable to create the lottie animation object from the JSON source")
             return
         }
@@ -249,7 +253,9 @@ class ContainerView: RCTView {
     }
 
     func play(fromFrame: AnimationFrameTime? = nil, toFrame: AnimationFrameTime) {
-        animationView?.play(fromFrame: fromFrame, toFrame: toFrame, loopMode: self.loop, completion: completionCallback)
+        animationView?.play(
+            fromFrame: fromFrame, toFrame: toFrame, loopMode: self.loop,
+            completion: completionCallback)
     }
 
     @objc func play() {
@@ -270,7 +276,7 @@ class ContainerView: RCTView {
     }
 
     // The animation view is a child of the RCTView, so if the bounds ever change, add those changes to the animation view as well
-    override func layoutSubviews() {
+    override open func layoutSubviews() {
         super.layoutSubviews()
         animationView?.frame = self.bounds
     }
@@ -319,7 +325,8 @@ class ContainerView: RCTView {
         if colorFilters.count > 0 {
             for filter in colorFilters {
                 guard let key = filter.value(forKey: "keypath") as? String,
-                      let platformColor = filter.value(forKey: "color") as? PlatformColor else { break }
+                    let platformColor = filter.value(forKey: "color") as? PlatformColor
+                else { break }
                 let keypath: String = "\(key).**.Color"
                 let fillKeypath = AnimationKeypath(keypath: keypath)
                 let colorFilterValueProvider = ColorValueProvider(platformColor.lottieColorValue)
@@ -347,7 +354,9 @@ class ContainerView: RCTView {
             guard let self = self else { return }
 
             if let error = error {
-                self.failureCallback("Unable to fetch the Lottie animation from the URL: \(error.localizedDescription)")
+                self.failureCallback(
+                    "Unable to fetch the Lottie animation from the URL: \(error.localizedDescription)"
+                )
                 return
             }
 
@@ -370,7 +379,9 @@ class ContainerView: RCTView {
                     self.replaceAnimationView(next: nextAnimationView)
                 }
             } catch {
-                self.failureCallback("Unable to decode the Lottie animation object from the fetched URL source: \(error.localizedDescription)")
+                self.failureCallback(
+                    "Unable to decode the Lottie animation object from the fetched URL source: \(error.localizedDescription)"
+                )
             }
         }.resume()
     }
