diff --git a/node_modules/metro-transform-worker/src/index.js b/node_modules/metro-transform-worker/src/index.js
index 27d4cb3..5ae4b1a 100644
--- a/node_modules/metro-transform-worker/src/index.js
+++ b/node_modules/metro-transform-worker/src/index.js
@@ -289,6 +289,8 @@ async function transformJS(file, { config, options, projectRoot }) {
         lineCount: countLines(code),
         map,
         functionMap: file.functionMap,
+        modules: file.babelMetadata?.modules,
+        ast: wrappedAst,
       },
       type: file.type,
     },
@@ -334,6 +336,9 @@ async function transformJSWithBabel(file, context) {
   const jsFile = {
     ...file,
     ast: transformResult.ast,
+
+    // Add the additional metadata which could have tree-shaking data.
+    babelMetadata: transformResult.metadata,
     functionMap:
       transformResult.metadata?.metro?.functionMap ??
       // Fallback to deprecated explicitly-generated `functionMap`
