diff --git a/RNWorklets.podspec b/RNWorklets.podspec
--- a/RNWorklets.podspec
+++ b/RNWorklets.podspec
@@ -56,6 +56,7 @@
   install_modules_dependencies(s)
 
   s.dependency 'React-jsi'
+  s.dependency 'React-jsitooling'
   s.dependency 'React-hermes'
   
   s.pod_target_xcconfig = {
@@ -70,6 +71,7 @@
       '"$(PODS_ROOT)/DoubleConversion"',
       '"$(PODS_ROOT)/Headers/Private/React-Core"',
       '"$(PODS_ROOT)/Headers/Private/Yoga"',
+      '"${PODS_CONFIGURATION_BUILD_DIR}/React-jsitooling/JSITooling.framework/Headers"',
     ].join(' '),
     "FRAMEWORK_SEARCH_PATHS" => '"${PODS_CONFIGURATION_BUILD_DIR}/React-hermes"',
     "CLANG_CXX_LANGUAGE_STANDARD" => "c++20",
diff --git a/compatibility.json b/compatibility.json
--- a/compatibility.json
+++ b/compatibility.json
@@ -1,9 +1,9 @@
 {
   "nightly": {
-    "react-native": ["0.83", "0.84", "0.85", "0.86"]
+    "react-native": ["0.83", "0.84", "0.85", "0.86", "0.87"]
   },
   "0.10.x": {
-    "react-native": ["0.83", "0.84", "0.85", "0.86"]
+    "react-native": ["0.83", "0.84", "0.85", "0.86", "0.87"]
   },
   "0.9.x": {
     "react-native": ["0.83", "0.84", "0.85", "0.86"]
