diff --git a/node_modules/react-native-pager-view/android/src/paper/java/com/reactnativepagerview/PagerViewViewManager.kt b/node_modules/react-native-pager-view/android/src/paper/java/com/reactnativepagerview/PagerViewViewManager.kt
index 959362c..8d8da1f 100644
--- a/node_modules/react-native-pager-view/android/src/paper/java/com/reactnativepagerview/PagerViewViewManager.kt
+++ b/node_modules/react-native-pager-view/android/src/paper/java/com/reactnativepagerview/PagerViewViewManager.kt
@@ -34,7 +34,7 @@ class PagerViewViewManager : ViewGroupManager<NestedScrollableHost>() {
         vp.adapter = ViewPagerAdapter()
         //https://github.com/callstack/react-native-viewpager/issues/183
         vp.isSaveEnabled = false
-        eventDispatcher = reactContext.getNativeModule(UIManagerModule::class.java)!!.eventDispatcher
+        eventDispatcher = reactContext.getNativeModule(UIManagerModule::class.java)!!.getEventDispatcher()
 
         vp.post {
             vp.registerOnPageChangeCallback(object : OnPageChangeCallback() {
