diff --git a/node_modules/react-native-reanimated/android/src/main/java/com/swmansion/reanimated/ReactNativeUtils.java b/node_modules/react-native-reanimated/android/src/main/java/com/swmansion/reanimated/ReactNativeUtils.java
index b15ef95..34d27f8 100644
--- a/node_modules/react-native-reanimated/android/src/main/java/com/swmansion/reanimated/ReactNativeUtils.java
+++ b/node_modules/react-native-reanimated/android/src/main/java/com/swmansion/reanimated/ReactNativeUtils.java
@@ -14,7 +14,7 @@ public class ReactNativeUtils {
     if (view.getBackground() != null) {
       Drawable background = view.getBackground();
       if (background instanceof ReactViewBackgroundDrawable) {
-        return ((ReactViewBackgroundDrawable) background).getFullBorderRadius();
+        return 0;
       }
     } else if (view instanceof ReactImageView) {
       try {
diff --git a/node_modules/react-native-reanimated/android/src/reactNativeVersionPatch/ReanimatedUIManager/latest/com/swmansion/reanimated/layoutReanimation/ReanimatedUIManager.java b/node_modules/react-native-reanimated/android/src/reactNativeVersionPatch/ReanimatedUIManager/latest/com/swmansion/reanimated/layoutReanimation/ReanimatedUIManager.java
index 256906f..9c84f87 100644
--- a/node_modules/react-native-reanimated/android/src/reactNativeVersionPatch/ReanimatedUIManager/latest/com/swmansion/reanimated/layoutReanimation/ReanimatedUIManager.java
+++ b/node_modules/react-native-reanimated/android/src/reactNativeVersionPatch/ReanimatedUIManager/latest/com/swmansion/reanimated/layoutReanimation/ReanimatedUIManager.java
@@ -125,22 +125,6 @@ public class ReanimatedUIManager extends UIManagerModule {
     super.measureLayout(tag, ancestorTag, errorCallback, successCallback);
   }
 
-  /**
-   * Like {@link #measure} and {@link #measureLayout} but measures relative to the immediate parent.
-   *
-   * <p>NB: Unlike {@link #measure}, this will measure relative to the view layout, not the visible
-   * window which can cause unexpected results when measuring relative to things like ScrollViews
-   * that can have offset content on the screen.
-   *
-   * @deprecated this method will not be available in FabricUIManager class.
-   */
-  @ReactMethod
-  @Deprecated
-  public void measureLayoutRelativeToParent(
-      int tag, Callback errorCallback, Callback successCallback) {
-    super.measureLayoutRelativeToParent(tag, errorCallback, successCallback);
-  }
-
   /**
    * Find the touch target child native view in the supplied root view hierarchy, given a react
    * target location.
@@ -186,26 +170,6 @@ public class ReanimatedUIManager extends UIManagerModule {
     super.dispatchViewManagerCommand(reactTag, commandId, commandArgs);
   }
 
-  /**
-   * Show a PopupMenu.
-   *
-   * @param reactTag the tag of the anchor view (the PopupMenu is displayed next to this view); this
-   *     needs to be the tag of a native view (shadow views can not be anchors)
-   * @param items the menu items as an array of strings
-   * @param error will be called if there is an error displaying the menu
-   * @param success will be called with the position of the selected item as the first argument, or
-   *     no arguments if the menu is dismissed
-   */
-  @ReactMethod
-  public void showPopupMenu(int reactTag, ReadableArray items, Callback error, Callback success) {
-    super.showPopupMenu(reactTag, items, error, success);
-  }
-
-  @ReactMethod
-  public void dismissPopupMenu() {
-    super.dismissPopupMenu();
-  }
-
   /**
    * LayoutAnimation API on Android is currently experimental. Therefore, it needs to be enabled
    * explicitly in order to avoid regression in existing application written for iOS using this API.
