diff --git a/node_modules/react-native-screens/RNScreens.podspec b/node_modules/react-native-screens/RNScreens.podspec
index 968d11b..1906c7c 100644
--- a/node_modules/react-native-screens/RNScreens.podspec
+++ b/node_modules/react-native-screens/RNScreens.podspec
@@ -48,28 +48,33 @@ Pod::Spec.new do |s|
   s.author       = { "author" => "author@domain.cn" }
   s.platforms    = { :ios => min_supported_ios_version, :tvos => min_supported_tvos_version, :visionos => min_supported_visionos_version }
   s.source       = { :git => "https://github.com/software-mansion/react-native-screens.git", :tag => "#{s.version}" }
-  s.source_files = source_files
-  s.project_header_files = "ios/bridging/Swift-Bridging.h"
   s.requires_arc = true
 
-  if !gamma_project_enabled
-    s.exclude_files = "ios/gamma/**/*.#{source_files_exts}"
-  else
-    s.exclude_files = "ios/stubs/**/*.#{source_files_exts}"
-    Pod::UI.puts "[RNScreens] Gamma project enabled. Including source files."
-  end
+  # Expo prebuilt xcframework support
+  if !Expo::PackagesConfig.instance.try_link_with_prebuilt_xcframework(s)
+    # Build from source
+    s.source_files = source_files
+    s.project_header_files = "ios/bridging/Swift-Bridging.h"
 
-  s.pod_target_xcconfig = rnscreens_config
+    if !gamma_project_enabled
+      s.exclude_files = "ios/gamma/**/*.#{source_files_exts}"
+    else
+      s.exclude_files = "ios/stubs/**/*.#{source_files_exts}"
+      Pod::UI.puts "[RNScreens] Gamma project enabled. Including source files."
+    end
 
-  install_modules_dependencies(s)
-  if new_arch_enabled
-    s.subspec "common" do |ss|
-      ss.source_files         = ["common/cpp/**/*.{cpp,h}", "cpp/**/*.{cpp,h}"]
-      ss.project_header_files = "common/cpp/**/*.h", "cpp/**/*.h" # Don't expose C++ headers publicly to allow importing framework into Swift files
-      ss.header_dir           = "rnscreens"
-      ss.pod_target_xcconfig  = { "HEADER_SEARCH_PATHS" => "\"$(PODS_TARGET_SRCROOT)/common/cpp\"" }
+    s.pod_target_xcconfig = rnscreens_config
+
+    if new_arch_enabled
+      s.subspec "common" do |ss|
+        ss.source_files         = ["common/cpp/**/*.{cpp,h}", "cpp/**/*.{cpp,h}"]
+        ss.project_header_files = "common/cpp/**/*.h", "cpp/**/*.h" # Don't expose C++ headers publicly to allow importing framework into Swift files
+        ss.header_dir           = "rnscreens"
+        ss.pod_target_xcconfig  = { "HEADER_SEARCH_PATHS" => "\"$(PODS_TARGET_SRCROOT)/common/cpp\"" }
+      end
     end
   end
 
+  install_modules_dependencies(s)
   s.dependency "React-RCTImage"
 end
