diff --git a/node_modules/react-native/Libraries/AppDelegate/RCTReactNativeFactory.mm b/node_modules/react-native/Libraries/AppDelegate/RCTReactNativeFactory.mm
index 75c963a..dbe3fa2 100644
--- a/node_modules/react-native/Libraries/AppDelegate/RCTReactNativeFactory.mm
+++ b/node_modules/react-native/Libraries/AppDelegate/RCTReactNativeFactory.mm
@@ -48,7 +48,11 @@ using namespace facebook::react;
 {
   if (self = [super init]) {
     self.delegate = delegate;
-    [self _setUpFeatureFlags];
+    try {
+      [self _setUpFeatureFlags];
+    } catch (...) {
+      // todo(chrfalch): This is untill we know how to create multiple React Native Factories
+    }

     auto newArchEnabled = [self newArchEnabled];
     auto fabricEnabled = [self fabricEnabled];
