diff --git a/node_modules/react-native/scripts/bundle.js b/node_modules/react-native/scripts/bundle.js
index 53acb05..9f4fe19 100644
--- a/node_modules/react-native/scripts/bundle.js
+++ b/node_modules/react-native/scripts/bundle.js
@@ -12,7 +12,7 @@
 
 const {bundleCommand: bc} = require('@react-native/community-cli-plugin');
 const {execSync} = require('child_process');
-const program = require('commander');
+const {program} = require('commander');
 const {existsSync, readFileSync} = require('fs');
 const path = require('path');
 
diff --git a/node_modules/react-native/scripts/react-native-xcode.sh b/node_modules/react-native/scripts/react-native-xcode.sh
index 0886282..1900c21 100755
--- a/node_modules/react-native/scripts/react-native-xcode.sh
+++ b/node_modules/react-native/scripts/react-native-xcode.sh
@@ -90,6 +90,7 @@ fi
 
 [ -z "$NODE_ARGS" ] && export NODE_ARGS=""
 
+[ -z "$CLI_PATH" ] && CLI_PATH="$REACT_NATIVE_DIR/scripts/bundle.js"
 [ -z "$BUNDLE_COMMAND" ] && BUNDLE_COMMAND="bundle"
 
 [ -z "$COMPOSE_SOURCEMAP_PATH" ] && COMPOSE_SOURCEMAP_PATH="$REACT_NATIVE_DIR/scripts/compose-source-maps.js"
@@ -147,7 +148,7 @@ else
 fi
 
 # shellcheck disable=SC2086
-"$NODE_BINARY" $NODE_ARGS "$REACT_NATIVE_DIR/scripts/bundle.js" \
+"$NODE_BINARY" $NODE_ARGS "$CLI_PATH" $BUNDLE_COMMAND \
   $CONFIG_ARG \
   --config-cmd "$CONFIG" \
   --entry-file "$ENTRY_FILE" \
