diff --git a/node_modules/react-native/src/private/specs/components/DebuggingOverlayNativeComponent.js b/node_modules/react-native/src/private/specs/components/DebuggingOverlayNativeComponent.js
index 1d2f230..624e434 100644
--- a/node_modules/react-native/src/private/specs/components/DebuggingOverlayNativeComponent.js
+++ b/node_modules/react-native/src/private/specs/components/DebuggingOverlayNativeComponent.js
@@ -35,14 +35,14 @@ export type ElementRectangle = {
 };
 
 interface NativeCommands {
-  +highlightTraceUpdates: (
-    viewRef: React.ElementRef<DebuggingOverlayNativeComponentType>,
-    updates: $ReadOnlyArray<TraceUpdate>,
-  ) => void;
-  +highlightElements: (
-    viewRef: React.ElementRef<DebuggingOverlayNativeComponentType>,
-    elements: $ReadOnlyArray<ElementRectangle>,
-  ) => void;
+  // +highlightTraceUpdates: (
+  //   viewRef: React.ElementRef<DebuggingOverlayNativeComponentType>,
+  //   updates: $ReadOnlyArray<TraceUpdate>,
+  // ) => void;
+  // +highlightElements: (
+  //   viewRef: React.ElementRef<DebuggingOverlayNativeComponentType>,
+  //   elements: $ReadOnlyArray<ElementRectangle>,
+  // ) => void;
   +clearElementsHighlights: (
     viewRef: React.ElementRef<DebuggingOverlayNativeComponentType>,
   ) => void;
@@ -50,8 +50,8 @@ interface NativeCommands {
 
 export const Commands: NativeCommands = codegenNativeCommands<NativeCommands>({
   supportedCommands: [
-    'highlightTraceUpdates',
-    'highlightElements',
+    // 'highlightTraceUpdates',
+    // 'highlightElements',
     'clearElementsHighlights',
   ],
 });
