diff --git a/node_modules/react-native/src/private/devsupport/rndevtools/setUpFuseboxReactDevToolsDispatcher.js b/node_modules/react-native/src/private/devsupport/rndevtools/setUpFuseboxReactDevToolsDispatcher.js
index 79f7f4d..3c85d2f 100644
--- a/node_modules/react-native/src/private/devsupport/rndevtools/setUpFuseboxReactDevToolsDispatcher.js
+++ b/node_modules/react-native/src/private/devsupport/rndevtools/setUpFuseboxReactDevToolsDispatcher.js
@@ -105,9 +105,11 @@ class FuseboxReactDevToolsDispatcher {
   }
 }
 
-Object.defineProperty(global, '__FUSEBOX_REACT_DEVTOOLS_DISPATCHER__', {
-  value: FuseboxReactDevToolsDispatcher,
-  configurable: false,
-  enumerable: false,
-  writable: false,
-});
+if (!('__FUSEBOX_REACT_DEVTOOLS_DISPATCHER__' in global)) {
+  Object.defineProperty(global, '__FUSEBOX_REACT_DEVTOOLS_DISPATCHER__', {
+    value: FuseboxReactDevToolsDispatcher,
+    configurable: false,
+    enumerable: false,
+    writable: false,
+  });
+}
