diff --git a/node_modules/react-native/Libraries/Core/setUpTimers.js b/node_modules/react-native/Libraries/Core/setUpTimers.js
index 7dc59d9..06ad68b 100644
--- a/node_modules/react-native/Libraries/Core/setUpTimers.js
+++ b/node_modules/react-native/Libraries/Core/setUpTimers.js
@@ -26,6 +26,15 @@ const isEventLoopEnabled = (() => {
     return false;
   }
 
+  const isRN76 = typeof NativeReactNativeFeatureFlags?.['useModernRuntimeScheduler'] === 'function'
+  if (isRN76) {
+    // RN 76 support for dev-menu: dev menu bundle is built with RN 77 and this code makes sure it uses 76 feature flags
+    return (
+      NativeReactNativeFeatureFlags.useModernRuntimeScheduler() &&
+      NativeReactNativeFeatureFlags.enableMicrotasks()
+    );
+  }
+
   return (
     ReactNativeFeatureFlags.enableBridgelessArchitecture() &&
     !ReactNativeFeatureFlags.disableEventLoopOnBridgeless()
