diff --git a/node_modules/react-native-svg/common/cpp/react/renderer/components/rnsvg/RNSVGLayoutableShadowNode.cpp b/node_modules/react-native-svg/common/cpp/react/renderer/components/rnsvg/RNSVGLayoutableShadowNode.cpp
index a42fde9..f980611 100644
--- a/node_modules/react-native-svg/common/cpp/react/renderer/components/rnsvg/RNSVGLayoutableShadowNode.cpp
+++ b/node_modules/react-native-svg/common/cpp/react/renderer/components/rnsvg/RNSVGLayoutableShadowNode.cpp
@@ -25,8 +25,8 @@ void RNSVGLayoutableShadowNode::setZeroDimensions() {
   // properties like width when viewBox scale is set.
   auto style = yogaNode_.style();
 #if REACT_NATIVE_MINOR_VERSION >= 77
-  style.setDimension(yoga::Dimension::Width, yoga::StyleLength::points(0));
-  style.setDimension(yoga::Dimension::Height, yoga::StyleLength::points(0));
+	style.setDimension(yoga::Dimension::Width, yoga::Style::SizeLength::points(0));
+  style.setDimension(yoga::Dimension::Height, yoga::Style::SizeLength::points(0));
 #else
   style.setDimension(yoga::Dimension::Width, yoga::value::points(0));
   style.setDimension(yoga::Dimension::Height, yoga::value::points(0));
