diff --git a/lib/commonjs/RNRenderer.js b/lib/commonjs/RNRenderer.js
deleted file mode 100644
index 7845103a88ce49f6cef0f68a904974c0f84da645..0000000000000000000000000000000000000000
diff --git a/lib/commonjs/RNRenderer.web.js b/lib/commonjs/RNRenderer.web.js
deleted file mode 100644
index a94a363d11db1fe2f104d4950aa1748496d4f295..0000000000000000000000000000000000000000
diff --git a/lib/commonjs/handlers/gestures/GestureDetector/useDetectorUpdater.js b/lib/commonjs/handlers/gestures/GestureDetector/useDetectorUpdater.js
index 27a4164f4082a465709f7d20e8d80c9e49bf849e..cda7cb38e79e6bc67d1e1341ad77196752d73c20 100644
--- a/lib/commonjs/handlers/gestures/GestureDetector/useDetectorUpdater.js
+++ b/lib/commonjs/handlers/gestures/GestureDetector/useDetectorUpdater.js
@@ -24,7 +24,6 @@ function useDetectorUpdater(state, preparedGesture, gesturesToAttach, gestureCon
     const viewTag = (0, _findNodeHandle.default)(state.viewRef);
     const didUnderlyingViewChange = viewTag !== state.previousViewTag;
     if (didUnderlyingViewChange || (0, _needsToReattach.needsToReattach)(preparedGesture, gesturesToAttach)) {
-      (0, _utils.validateDetectorChildren)(state.viewRef);
       (0, _dropHandlers.dropHandlers)(preparedGesture);
       (0, _attachHandlers.attachHandlers)({
         preparedGesture,
diff --git a/lib/commonjs/handlers/gestures/GestureDetector/utils.js b/lib/commonjs/handlers/gestures/GestureDetector/utils.js
index 28412d948d51b46566ad0fc97049603b79b81200..78c565caac91da4f773a2baf371d6da9b7045d66 100644
--- a/lib/commonjs/handlers/gestures/GestureDetector/utils.js
+++ b/lib/commonjs/handlers/gestures/GestureDetector/utils.js
@@ -8,8 +8,6 @@ exports.checkGestureCallbacksForWorklets = checkGestureCallbacksForWorklets;
 exports.extractGestureRelations = extractGestureRelations;
 exports.useForceRender = useForceRender;
 exports.useWebEventHandlers = useWebEventHandlers;
-exports.validateDetectorChildren = validateDetectorChildren;
-var _reactNative = require("react-native");
 var _utils = require("../../../utils");
 var _gesture = require("../gesture");
 var _FlingGestureHandler = require("../../FlingGestureHandler");
@@ -21,7 +19,6 @@ var _hoverGesture = require("../hoverGesture");
 var _NativeViewGestureHandler = require("../../NativeViewGestureHandler");
 var _gestureHandlerCommon = require("../../gestureHandlerCommon");
 var _EnableNewWebImplementation = require("../../../EnableNewWebImplementation");
-var _RNRenderer = require("../../../RNRenderer");
 var _react = require("react");
 var _reanimatedWrapper = require("../reanimatedWrapper");
 var _eventReceiver = require("../eventReceiver");
@@ -88,51 +85,6 @@ function checkGestureCallbacksForWorklets(gesture) {
     console.warn((0, _utils.tagMessage)(`None of the callbacks in the gesture are worklets. If you wish to run them on the JS thread use '.runOnJS(true)' modifier on the gesture to make this explicit. Otherwise, mark the callbacks as 'worklet' to run them on the UI thread.`));
   }
 }
-
-// eslint-disable-next-line @typescript-eslint/no-explicit-any
-function validateDetectorChildren(ref) {
-  // Finds the first native view under the Wrap component and traverses the fiber tree upwards
-  // to check whether there is more than one native view as a pseudo-direct child of GestureDetector
-  // i.e. this is not ok:
-  //            Wrap
-  //             |
-  //            / \
-  //           /   \
-  //          /     \
-  //         /       \
-  //   NativeView  NativeView
-  //
-  // but this is fine:
-  //            Wrap
-  //             |
-  //         NativeView
-  //             |
-  //            / \
-  //           /   \
-  //          /     \
-  //         /       \
-  //   NativeView  NativeView
-  if (__DEV__ && _reactNative.Platform.OS !== 'web') {
-    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
-    const wrapType =
-    // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
-    ref._reactInternals.elementType;
-
-    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
-    let instance = _RNRenderer.RNRenderer.findHostInstance_DEPRECATED(ref)._internalFiberInstanceHandleDEV;
-
-    // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
-    while (instance && instance.elementType !== wrapType) {
-      // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
-      if (instance.sibling) {
-        throw new Error('GestureDetector has more than one native view as its children. This can happen if you are using a custom component that renders multiple views, like React.Fragment. You should wrap content of GestureDetector with a <View> or <Animated.View>.');
-      }
-
-      // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-member-access
-      instance = instance.return;
-    }
-  }
-}
 function useForceRender() {
   const [renderState, setRenderState] = (0, _react.useState)(false);
   const forceRender = (0, _react.useCallback)(() => {
diff --git a/lib/module/RNRenderer.js b/lib/module/RNRenderer.js
deleted file mode 100644
index be7999d6237748e3834183835c3e2a064841c8db..0000000000000000000000000000000000000000
diff --git a/lib/module/RNRenderer.web.js b/lib/module/RNRenderer.web.js
deleted file mode 100644
index 5d9872ab168754600944fad432f278bb674ccaa6..0000000000000000000000000000000000000000
diff --git a/lib/module/handlers/gestures/GestureDetector/useDetectorUpdater.js b/lib/module/handlers/gestures/GestureDetector/useDetectorUpdater.js
index c8e3ae2f4924b3ecf7b117914b502f995b8dd2fb..1fd877a6adf77ed17edbe79c301eda553c4a2c45 100644
--- a/lib/module/handlers/gestures/GestureDetector/useDetectorUpdater.js
+++ b/lib/module/handlers/gestures/GestureDetector/useDetectorUpdater.js
@@ -5,7 +5,7 @@ import { attachHandlers } from './attachHandlers';
 import { updateHandlers } from './updateHandlers';
 import { needsToReattach } from './needsToReattach';
 import { dropHandlers } from './dropHandlers';
-import { useForceRender, validateDetectorChildren } from './utils';
+import { useForceRender } from './utils';
 import findNodeHandle from '../../../findNodeHandle';
 
 // Returns a function that's responsible for updating the attached gestures
@@ -20,7 +20,6 @@ export function useDetectorUpdater(state, preparedGesture, gesturesToAttach, ges
     const viewTag = findNodeHandle(state.viewRef);
     const didUnderlyingViewChange = viewTag !== state.previousViewTag;
     if (didUnderlyingViewChange || needsToReattach(preparedGesture, gesturesToAttach)) {
-      validateDetectorChildren(state.viewRef);
       dropHandlers(preparedGesture);
       attachHandlers({
         preparedGesture,
diff --git a/lib/module/handlers/gestures/GestureDetector/utils.js b/lib/module/handlers/gestures/GestureDetector/utils.js
index 9c752528da231b78c8bf0a37bf5885b47434755c..587dee50386d5459332a2955b071ed83b32202d5 100644
--- a/lib/module/handlers/gestures/GestureDetector/utils.js
+++ b/lib/module/handlers/gestures/GestureDetector/utils.js
@@ -1,6 +1,5 @@
 "use strict";
 
-import { Platform } from 'react-native';
 import { isTestEnv, tagMessage } from '../../../utils';
 import { BaseGesture } from '../gesture';
 import { flingGestureHandlerProps } from '../../FlingGestureHandler';
@@ -12,7 +11,6 @@ import { hoverGestureHandlerProps } from '../hoverGesture';
 import { nativeViewGestureHandlerProps } from '../../NativeViewGestureHandler';
 import { baseGestureHandlerWithDetectorProps } from '../../gestureHandlerCommon';
 import { isNewWebImplementationEnabled } from '../../../EnableNewWebImplementation';
-import { RNRenderer } from '../../../RNRenderer';
 import { useCallback, useRef, useState } from 'react';
 import { Reanimated } from '../reanimatedWrapper';
 import { onGestureHandlerEvent } from '../eventReceiver';
@@ -79,51 +77,6 @@ export function checkGestureCallbacksForWorklets(gesture) {
     console.warn(tagMessage(`None of the callbacks in the gesture are worklets. If you wish to run them on the JS thread use '.runOnJS(true)' modifier on the gesture to make this explicit. Otherwise, mark the callbacks as 'worklet' to run them on the UI thread.`));
   }
 }
-
-// eslint-disable-next-line @typescript-eslint/no-explicit-any
-export function validateDetectorChildren(ref) {
-  // Finds the first native view under the Wrap component and traverses the fiber tree upwards
-  // to check whether there is more than one native view as a pseudo-direct child of GestureDetector
-  // i.e. this is not ok:
-  //            Wrap
-  //             |
-  //            / \
-  //           /   \
-  //          /     \
-  //         /       \
-  //   NativeView  NativeView
-  //
-  // but this is fine:
-  //            Wrap
-  //             |
-  //         NativeView
-  //             |
-  //            / \
-  //           /   \
-  //          /     \
-  //         /       \
-  //   NativeView  NativeView
-  if (__DEV__ && Platform.OS !== 'web') {
-    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
-    const wrapType =
-    // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
-    ref._reactInternals.elementType;
-
-    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
-    let instance = RNRenderer.findHostInstance_DEPRECATED(ref)._internalFiberInstanceHandleDEV;
-
-    // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
-    while (instance && instance.elementType !== wrapType) {
-      // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
-      if (instance.sibling) {
-        throw new Error('GestureDetector has more than one native view as its children. This can happen if you are using a custom component that renders multiple views, like React.Fragment. You should wrap content of GestureDetector with a <View> or <Animated.View>.');
-      }
-
-      // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-member-access
-      instance = instance.return;
-    }
-  }
-}
 export function useForceRender() {
   const [renderState, setRenderState] = useState(false);
   const forceRender = useCallback(() => {
diff --git a/src/RNRenderer.ts b/src/RNRenderer.ts
deleted file mode 100644
index 7a585b64c0bc7167718e187e96b29ad8c7cbff17..0000000000000000000000000000000000000000
diff --git a/src/RNRenderer.web.ts b/src/RNRenderer.web.ts
deleted file mode 100644
index d46b825958bf5123babebee49ea95664fbe6af43..0000000000000000000000000000000000000000
diff --git a/src/handlers/gestures/GestureDetector/useDetectorUpdater.ts b/src/handlers/gestures/GestureDetector/useDetectorUpdater.ts
index d6d82c4b70c781732831d12815d84e2ac685478e..34fc83d99b39ec6df9fd002bb7121c6c348dffec 100644
--- a/src/handlers/gestures/GestureDetector/useDetectorUpdater.ts
+++ b/src/handlers/gestures/GestureDetector/useDetectorUpdater.ts
@@ -11,7 +11,7 @@ import { attachHandlers } from './attachHandlers';
 import { updateHandlers } from './updateHandlers';
 import { needsToReattach } from './needsToReattach';
 import { dropHandlers } from './dropHandlers';
-import { useForceRender, validateDetectorChildren } from './utils';
+import { useForceRender } from './utils';
 import findNodeHandle from '../../../findNodeHandle';
 
 // Returns a function that's responsible for updating the attached gestures
@@ -36,7 +36,6 @@ export function useDetectorUpdater(
         didUnderlyingViewChange ||
         needsToReattach(preparedGesture, gesturesToAttach)
       ) {
-        validateDetectorChildren(state.viewRef);
         dropHandlers(preparedGesture);
         attachHandlers({
           preparedGesture,
diff --git a/src/handlers/gestures/GestureDetector/utils.ts b/src/handlers/gestures/GestureDetector/utils.ts
index fc713189b2e3401df7054b0495649b290743987f..9f9f20a6fbeb38df966848a4bf071dca305108aa 100644
--- a/src/handlers/gestures/GestureDetector/utils.ts
+++ b/src/handlers/gestures/GestureDetector/utils.ts
@@ -1,5 +1,3 @@
-import { Platform } from 'react-native';
-
 import { isTestEnv, tagMessage } from '../../../utils';
 import { GestureRef, BaseGesture, GestureType } from '../gesture';
 
@@ -18,7 +16,6 @@ import {
   baseGestureHandlerWithDetectorProps,
 } from '../../gestureHandlerCommon';
 import { isNewWebImplementationEnabled } from '../../../EnableNewWebImplementation';
-import { RNRenderer } from '../../../RNRenderer';
 import { useCallback, useRef, useState } from 'react';
 import { Reanimated } from '../reanimatedWrapper';
 import { onGestureHandlerEvent } from '../eventReceiver';
@@ -124,56 +121,6 @@ export function checkGestureCallbacksForWorklets(gesture: GestureType) {
   }
 }
 
-// eslint-disable-next-line @typescript-eslint/no-explicit-any
-export function validateDetectorChildren(ref: any) {
-  // Finds the first native view under the Wrap component and traverses the fiber tree upwards
-  // to check whether there is more than one native view as a pseudo-direct child of GestureDetector
-  // i.e. this is not ok:
-  //            Wrap
-  //             |
-  //            / \
-  //           /   \
-  //          /     \
-  //         /       \
-  //   NativeView  NativeView
-  //
-  // but this is fine:
-  //            Wrap
-  //             |
-  //         NativeView
-  //             |
-  //            / \
-  //           /   \
-  //          /     \
-  //         /       \
-  //   NativeView  NativeView
-  if (__DEV__ && Platform.OS !== 'web') {
-    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
-    const wrapType =
-      // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
-      ref._reactInternals.elementType;
-
-    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
-    let instance =
-      RNRenderer.findHostInstance_DEPRECATED(
-        ref
-      )._internalFiberInstanceHandleDEV;
-
-    // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
-    while (instance && instance.elementType !== wrapType) {
-      // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
-      if (instance.sibling) {
-        throw new Error(
-          'GestureDetector has more than one native view as its children. This can happen if you are using a custom component that renders multiple views, like React.Fragment. You should wrap content of GestureDetector with a <View> or <Animated.View>.'
-        );
-      }
-
-      // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-member-access
-      instance = instance.return;
-    }
-  }
-}
-
 export function useForceRender() {
   const [renderState, setRenderState] = useState(false);
   const forceRender = useCallback(() => {
