diff --git a/dist/cjs/exports/AccessibilityInfo/index.js b/dist/cjs/exports/AccessibilityInfo/index.js
index 30b4c2a..caf474e 100644
--- a/dist/cjs/exports/AccessibilityInfo/index.js
+++ b/dist/cjs/exports/AccessibilityInfo/index.js
@@ -15,7 +15,7 @@ var _canUseDom = _interopRequireDefault(require("../../modules/canUseDom"));
 function isScreenReaderEnabled() {
   return Promise.resolve(true);
 }
-const prefersReducedMotionMedia = _canUseDom.default ? window.matchMedia('(prefers-reduced-motion: reduce)') : null;
+const prefersReducedMotionMedia = _canUseDom.default && typeof window.matchMedia === 'function' ? window.matchMedia('(prefers-reduced-motion: reduce)') : null;
 const handlers = new Map();
 const AccessibilityInfo = {
   /**
diff --git a/dist/cjs/exports/Appearance/index.js b/dist/cjs/exports/Appearance/index.js
index b79a4a5..9e95cd8 100644
--- a/dist/cjs/exports/Appearance/index.js
+++ b/dist/cjs/exports/Appearance/index.js
@@ -14,7 +14,7 @@ exports.__esModule = true;
 exports.default = void 0;
 var _canUseDom = _interopRequireDefault(require("../../modules/canUseDom"));
 function getQuery() {
-  return _canUseDom.default ? window.matchMedia('(prefers-color-scheme: dark)') : null;
+  return _canUseDom.default && typeof window.matchMedia === 'function' ? window.matchMedia('(prefers-color-scheme: dark)') : null;
 }
 const query = getQuery();
 const listenerMapping = new WeakMap();
diff --git a/dist/cjs/modules/useElementLayout/index.js b/dist/cjs/modules/useElementLayout/index.js
index 83bec9a..4ce98c1 100644
--- a/dist/cjs/modules/useElementLayout/index.js
+++ b/dist/cjs/modules/useElementLayout/index.js
@@ -16,7 +16,7 @@ var _useLayoutEffect = _interopRequireDefault(require("../useLayoutEffect"));
 const DOM_LAYOUT_HANDLER_NAME = '__reactLayoutHandler';
 let resizeObserver = null;
 function getResizeObserver() {
-  if (_canUseDom.default && resizeObserver == null) {
+  if (_canUseDom.default && typeof window.ResizeObserver === 'function' && resizeObserver == null) {
     resizeObserver = new window.ResizeObserver(function (entries) {
       entries.forEach(entry => {
         const node = entry.target;
diff --git a/dist/cjs/vendor/react-native/VirtualizedList/index.js b/dist/cjs/vendor/react-native/VirtualizedList/index.js
index 3785340..4bbf782 100644
--- a/dist/cjs/vendor/react-native/VirtualizedList/index.js
+++ b/dist/cjs/vendor/react-native/VirtualizedList/index.js
@@ -312,8 +312,6 @@ class VirtualizedList extends _StateSafePureComponent.default {
   _unregisterAsNestedChild = childList => {
     this._nestedChildLists.remove(childList.ref);
   };
-  state;
-
   // REACT-NATIVE-WEB patch to preserve during future RN merges: Support inverted wheel scroller.
   invertedWheelEventHandler;
   constructor(props) {
diff --git a/dist/exports/AccessibilityInfo/index.js b/dist/exports/AccessibilityInfo/index.js
index 385246e..749796c 100644
--- a/dist/exports/AccessibilityInfo/index.js
+++ b/dist/exports/AccessibilityInfo/index.js
@@ -11,7 +11,7 @@ import canUseDOM from '../../modules/canUseDom';
 function isScreenReaderEnabled() {
   return Promise.resolve(true);
 }
-const prefersReducedMotionMedia = canUseDOM ? window.matchMedia('(prefers-reduced-motion: reduce)') : null;
+const prefersReducedMotionMedia = canUseDOM && typeof window.matchMedia === 'function' ? window.matchMedia('(prefers-reduced-motion: reduce)') : null;
 const handlers = new Map();
 const AccessibilityInfo = {
   /**
diff --git a/dist/exports/Appearance/index.js b/dist/exports/Appearance/index.js
index 3a25f5c..34071c7 100644
--- a/dist/exports/Appearance/index.js
+++ b/dist/exports/Appearance/index.js
@@ -10,7 +10,7 @@
 
 import canUseDOM from '../../modules/canUseDom';
 function getQuery() {
-  return canUseDOM ? window.matchMedia('(prefers-color-scheme: dark)') : null;
+  return canUseDOM && typeof window.matchMedia === 'function' ? window.matchMedia('(prefers-color-scheme: dark)') : null;
 }
 const query = getQuery();
 const listenerMapping = new WeakMap();
diff --git a/dist/modules/useElementLayout/index.js b/dist/modules/useElementLayout/index.js
index b0be9a0..849c0df 100644
--- a/dist/modules/useElementLayout/index.js
+++ b/dist/modules/useElementLayout/index.js
@@ -11,7 +11,7 @@ import useLayoutEffect from '../useLayoutEffect';
 const DOM_LAYOUT_HANDLER_NAME = '__reactLayoutHandler';
 let resizeObserver = null;
 function getResizeObserver() {
-  if (canUseDOM && resizeObserver == null) {
+  if (canUseDOM && typeof window.ResizeObserver === 'function' && resizeObserver == null) {
     resizeObserver = new window.ResizeObserver(function (entries) {
       entries.forEach(entry => {
         const node = entry.target;
diff --git a/dist/vendor/react-native/VirtualizedList/index.js b/dist/vendor/react-native/VirtualizedList/index.js
index 0061999..8498c8f 100644
--- a/dist/vendor/react-native/VirtualizedList/index.js
+++ b/dist/vendor/react-native/VirtualizedList/index.js
@@ -306,8 +306,6 @@ class VirtualizedList extends StateSafePureComponent {
   _unregisterAsNestedChild = childList => {
     this._nestedChildLists.remove(childList.ref);
   };
-  state;
-
   // REACT-NATIVE-WEB patch to preserve during future RN merges: Support inverted wheel scroller.
   invertedWheelEventHandler;
   constructor(props) {
diff --git a/src/exports/AccessibilityInfo/index.ts b/src/exports/AccessibilityInfo/index.ts
index 03ad60e..a3d5579 100644
--- a/src/exports/AccessibilityInfo/index.ts
+++ b/src/exports/AccessibilityInfo/index.ts
@@ -16,9 +16,10 @@ function isScreenReaderEnabled(): Promise<boolean> {
   return Promise.resolve(true);
 }
 
-const prefersReducedMotionMedia = canUseDOM
-  ? window.matchMedia('(prefers-reduced-motion: reduce)')
-  : null;
+const prefersReducedMotionMedia =
+  canUseDOM && typeof window.matchMedia === 'function'
+    ? window.matchMedia('(prefers-reduced-motion: reduce)')
+    : null;
 
 const handlers = new Map<ChangeHandler, DOMChangeListener>();
 
diff --git a/src/exports/Appearance/index.ts b/src/exports/Appearance/index.ts
index 69b1d30..a80a9b1 100644
--- a/src/exports/Appearance/index.ts
+++ b/src/exports/Appearance/index.ts
@@ -20,7 +20,9 @@ type AppearanceListener = (preferences: AppearancePreferences) => void;
 type DOMAppearanceListener = (ev: MediaQueryListEvent) => void;
 
 function getQuery(): MediaQueryList | null {
-  return canUseDOM ? window.matchMedia('(prefers-color-scheme: dark)') : null;
+  return canUseDOM && typeof window.matchMedia === 'function'
+    ? window.matchMedia('(prefers-color-scheme: dark)')
+    : null;
 }
 
 const query = getQuery();
diff --git a/src/modules/useElementLayout/index.ts b/src/modules/useElementLayout/index.ts
index ddd4987..1eadc86 100644
--- a/src/modules/useElementLayout/index.ts
+++ b/src/modules/useElementLayout/index.ts
@@ -20,7 +20,11 @@ type LayoutHandlerNode = HTMLElement & {
 let resizeObserver: Nullable<ResizeObserver> = null;
 
 function getResizeObserver(): Nullable<ResizeObserver> {
-  if (canUseDOM && resizeObserver == null) {
+  if (
+    canUseDOM &&
+    typeof window.ResizeObserver === 'function' &&
+    resizeObserver == null
+  ) {
     resizeObserver = new window.ResizeObserver(function (entries) {
       entries.forEach((entry) => {
         const node = entry.target as LayoutHandlerNode;
diff --git a/src/vendor/react-native/VirtualizedList/index.tsx b/src/vendor/react-native/VirtualizedList/index.tsx
index 8ef2ea7..089c66d 100644
--- a/src/vendor/react-native/VirtualizedList/index.tsx
+++ b/src/vendor/react-native/VirtualizedList/index.tsx
@@ -441,7 +441,7 @@ class VirtualizedList extends StateSafePureComponent<Props, State> {
     this._nestedChildLists.remove(childList.ref);
   };
 
-  state: State;
+  declare state: State;
 
   // REACT-NATIVE-WEB patch to preserve during future RN merges: Support inverted wheel scroller.
   invertedWheelEventHandler: Nullable<(ev: InvertedWheelEvent) => void>;
