diff --git a/node_modules/react-native-svg/android/src/main/java/com/horcrux/svg/VirtualView.java b/node_modules/react-native-svg/android/src/main/java/com/horcrux/svg/VirtualView.java
index 4d71806..771e89e 100644
--- a/node_modules/react-native-svg/android/src/main/java/com/horcrux/svg/VirtualView.java
+++ b/node_modules/react-native-svg/android/src/main/java/com/horcrux/svg/VirtualView.java
@@ -608,17 +608,19 @@ public abstract class VirtualView extends ReactViewGroup {
         setBottom(Math.min(bottom, root.getHeight()));
       }
     }
-    EventDispatcher eventDispatcher =
-        UIManagerHelper.getEventDispatcherForReactTag(mContext, getId());
-    if (eventDispatcher != null) {
-      eventDispatcher.dispatchEvent(
-        new SvgOnLayoutEvent(
-          UIManagerHelper.getSurfaceId(VirtualView.this),
-          this.getId(),
-          left,
-          top,
-          width,
-          height));
+    if (!mContext.isBridgeless()) {
+      EventDispatcher eventDispatcher =
+          UIManagerHelper.getEventDispatcherForReactTag(mContext, getId());
+      if (eventDispatcher != null) {
+        eventDispatcher.dispatchEvent(
+          new SvgOnLayoutEvent(
+            UIManagerHelper.getSurfaceId(VirtualView.this),
+            this.getId(),
+            left,
+            top,
+            width,
+            height));
+      }
     }
   }
 
