diff --git a/node_modules/react-native-webview/apple/RNCWKProcessPoolManager.h b/node_modules/react-native-webview/apple/RNCWKProcessPoolManager.h
index 6d2d215..49ad209 100644
--- a/node_modules/react-native-webview/apple/RNCWKProcessPoolManager.h
+++ b/node_modules/react-native-webview/apple/RNCWKProcessPoolManager.h
@@ -10,6 +10,6 @@
 @interface RNCWKProcessPoolManager : NSObject
 
 + (instancetype) sharedManager;
-- (WKProcessPool *)sharedProcessPool;
+- (WKProcessPool *)sharedProcessPoolForScopeKey:(NSString *)scopeKey;
 
 @end
diff --git a/node_modules/react-native-webview/apple/RNCWKProcessPoolManager.m b/node_modules/react-native-webview/apple/RNCWKProcessPoolManager.m
index 2b38ffe..4dd8717 100644
--- a/node_modules/react-native-webview/apple/RNCWKProcessPoolManager.m
+++ b/node_modules/react-native-webview/apple/RNCWKProcessPoolManager.m
@@ -10,11 +10,31 @@
 
 @interface RNCWKProcessPoolManager() {
   WKProcessPool *_sharedProcessPool;
+  NSMutableDictionary<NSString *, WKProcessPool *> *_pools;
 }
 @end
 
 @implementation RNCWKProcessPoolManager
 
+- (instancetype)init
+{
+  if (self = [super init]) {
+    _pools = [NSMutableDictionary new];
+  }
+  return self;
+}
+
+- (WKProcessPool *)sharedProcessPoolForScopeKey:(NSString *)scopeKey
+{
+  if (!scopeKey) {
+    return [self sharedProcessPool];
+  }
+  if (!_pools[scopeKey]) {
+    _pools[scopeKey] = [[WKProcessPool alloc] init];
+  }
+  return _pools[scopeKey];
+}
+
 + (id) sharedManager {
   static RNCWKProcessPoolManager *_sharedManager = nil;
   @synchronized(self) {
diff --git a/node_modules/react-native-webview/apple/RNCWebViewImpl.h b/node_modules/react-native-webview/apple/RNCWebViewImpl.h
index e19fddd..dfb94a6 100644
--- a/node_modules/react-native-webview/apple/RNCWebViewImpl.h
+++ b/node_modules/react-native-webview/apple/RNCWebViewImpl.h
@@ -46,6 +46,7 @@ shouldStartLoadForRequest:(NSMutableDictionary<NSString *, id> *)request
 
 #if !TARGET_OS_OSX
 @interface RNCWebViewImpl : RCTView <UIEditMenuInteractionDelegate>
+@property (nonatomic, strong) NSString *scopeKey;
 
 @property (nonatomic, nullable) UIEditMenuInteraction *editMenuInteraction API_AVAILABLE(ios(16.0));
 #else
diff --git a/node_modules/react-native-webview/apple/RNCWebViewImpl.m b/node_modules/react-native-webview/apple/RNCWebViewImpl.m
index 9726fd3..662c86b 100644
--- a/node_modules/react-native-webview/apple/RNCWebViewImpl.m
+++ b/node_modules/react-native-webview/apple/RNCWebViewImpl.m
@@ -457,7 +457,7 @@ RCTAutoInsetsProtocol>
     wkWebViewConfig.websiteDataStore = [WKWebsiteDataStore defaultDataStore];
   }
   if(self.useSharedProcessPool) {
-    wkWebViewConfig.processPool = [[RNCWKProcessPoolManager sharedManager] sharedProcessPool];
+    wkWebViewConfig.processPool = [[RNCWKProcessPoolManager sharedManager] sharedProcessPoolForScopeKey:self.scopeKey];
   }
   wkWebViewConfig.userContentController = [WKUserContentController new];
 
diff --git a/node_modules/react-native-webview/apple/RNCWebViewManager.mm b/node_modules/react-native-webview/apple/RNCWebViewManager.mm
index a0e0d00..4fa8c71 100644
--- a/node_modules/react-native-webview/apple/RNCWebViewManager.mm
+++ b/node_modules/react-native-webview/apple/RNCWebViewManager.mm
@@ -32,12 +32,27 @@ RCT_ENUM_CONVERTER(RNCWebViewPermissionGrantType, (@{
 @end
 
 @implementation RNCWebViewManager
+NSString *_scopeKey;
 
 RCT_EXPORT_MODULE(RNCWebView)
 
+- (instancetype)initWithExperienceStableLegacyId:(NSString *)experienceStableLegacyId
+                                        scopeKey:(NSString *)scopeKey
+                                    easProjectId:(NSString *)easProjectId
+                           kernelServiceDelegate:(id)kernelServiceInstance
+                                          params:(NSDictionary *)params
+{
+  if (self = [super init]) {
+    _scopeKey = scopeKey;
+  }
+  return self;
+}
+
 - (RNCView *)view
 {
-  return [[RNCWebViewImpl alloc] init];
+  RNCWebViewImpl *webview = [[RNCWebViewImpl alloc] init];
+  webview.scopeKey = _scopeKey;
+  return webview;
 }
 
 RCT_EXPORT_VIEW_PROPERTY(source, NSDictionary)
