diff --git a/node_modules/react-native-worklets/android/CMakeLists.txt b/node_modules/react-native-worklets/android/CMakeLists.txt
index 1d949d6..5e60310 100644
--- a/node_modules/react-native-worklets/android/CMakeLists.txt
+++ b/node_modules/react-native-worklets/android/CMakeLists.txt
@@ -105,7 +105,7 @@ else()
 endif()
 
 if(${JS_RUNTIME} STREQUAL "hermes")
-  target_link_libraries(worklets hermes-engine::libhermes)
+  target_link_libraries(worklets hermes-engine::hermesvm)
 
   if(${HERMES_ENABLE_DEBUGGER})
     string(APPEND CMAKE_CXX_FLAGS " -DHERMES_ENABLE_DEBUGGER=1")
diff --git a/node_modules/react-native-worklets/compatibility.json b/node_modules/react-native-worklets/compatibility.json
index 523807f..012dd2a 100644
--- a/node_modules/react-native-worklets/compatibility.json
+++ b/node_modules/react-native-worklets/compatibility.json
@@ -3,7 +3,7 @@
     "react-native": ["0.78", "0.79", "0.80", "0.81"]
   },
   "0.5.x": {
-    "react-native": ["0.78", "0.79", "0.80", "0.81"]
+    "react-native": ["0.78", "0.79", "0.80", "0.81", "0.82"]
   },
   "0.4.x": {
     "react-native": ["0.78", "0.79", "0.80", "0.81"]
