diff --git a/node_modules/react-native/ReactCommon/react/runtime/platform/ios/ReactCommon/RCTHost.mm b/node_modules/react-native/ReactCommon/react/runtime/platform/ios/ReactCommon/RCTHost.mm
index 2461633..a7ca382 100644
--- a/node_modules/react-native/ReactCommon/react/runtime/platform/ios/ReactCommon/RCTHost.mm
+++ b/node_modules/react-native/ReactCommon/react/runtime/platform/ios/ReactCommon/RCTHost.mm
@@ -334,11 +334,32 @@ class RCTHostHostTargetDelegate : public facebook::react::jsinspector_modern::Ho
   return YES;
 }
 
+
 - (void)instance:(RCTInstance *)instance didInitializeRuntime:(facebook::jsi::Runtime &)runtime
 {
   [self.runtimeDelegate host:self didInitializeRuntime:runtime];
+    
+  NSString *builtinsPath = [[NSBundle mainBundle] pathForResource:@"builtins" ofType:@"hbc"];
+  if (builtinsPath) {
+    NSData *builtinsData = [NSData dataWithContentsOfFile:builtinsPath];
+    if (builtinsData) {
+      auto buffer = std::make_shared<facebook::jsi::StringBuffer>(std::string((const char*)builtinsData.bytes, builtinsData.length));
+      runtime.evaluateJavaScript(buffer, "builtins.hbc");
+    }
+  }
+    // NSString *builtinsPath = [[NSBundle mainBundle] pathForResource:@"builtins" ofType:@"js"];
+    // if (builtinsPath) {
+    //   NSString *builtinsCode = [NSString stringWithContentsOfFile:builtinsPath encoding:NSUTF8StringEncoding error:nil];
+    //   if (builtinsCode) {
+
+    //       std::string builtinsString = [builtinsCode UTF8String];
+    //       auto buffer = std::make_shared<facebook::jsi::StringBuffer>(std::move(builtinsString));
+    //       runtime.evaluateJavaScript(buffer, "builtins.js");
+    //   }
+    // }
 }
 
+
 - (void)loadBundleAtURL:(NSURL *)sourceURL
              onProgress:(RCTSourceLoadProgressBlock)onProgress
              onComplete:(RCTSourceLoadBlock)loadCallback
