diff --git a/node_modules/react-native-gesture-handler/apple/Handlers/RNFlingHandler.h b/node_modules/react-native-gesture-handler/apple/Handlers/RNFlingHandler.h
index f5d37d6..6e3e3d5 100644
--- a/node_modules/react-native-gesture-handler/apple/Handlers/RNFlingHandler.h
+++ b/node_modules/react-native-gesture-handler/apple/Handlers/RNFlingHandler.h
@@ -1,4 +1,9 @@
+#if __has_include("../RNGHVector.h")
 #import "../RNGHVector.h"
+#else
+#import "RNGHVector.h"
+#endif
+
 #import "RNGestureHandler.h"
 
 @interface RNFlingGestureHandler : RNGestureHandler
diff --git a/node_modules/react-native-gesture-handler/apple/Handlers/RNHoverHandler.m b/node_modules/react-native-gesture-handler/apple/Handlers/RNHoverHandler.m
index f800c84..73199a5 100644
--- a/node_modules/react-native-gesture-handler/apple/Handlers/RNHoverHandler.m
+++ b/node_modules/react-native-gesture-handler/apple/Handlers/RNHoverHandler.m
@@ -148,7 +148,10 @@ API_AVAILABLE(ios(13.4))
 #if CHECK_TARGET(13_4)
   if (@available(iOS 13.4, *)) {
     RNBetterHoverGestureRecognizer *recognizer = (RNBetterHoverGestureRecognizer *)_recognizer;
-    APPLY_INT_PROP(hoverEffect);
+    id value = config[@"hoverEffect"];
+    if (value != nil) {
+      recognizer.hoverEffect = (RNGestureHandlerHoverEffect)[RCTConvert NSInteger:value];
+    }
   }
 #endif
 }
