diff --git a/node_modules/react-native-svg/src/index.d.ts b/node_modules/react-native-svg/src/index.d.ts
index 891a99d..8823106 100644
--- a/node_modules/react-native-svg/src/index.d.ts
+++ b/node_modules/react-native-svg/src/index.d.ts
@@ -231,7 +231,9 @@ export interface CommonPathProps
     TouchableProps,
     DefinitionProps,
     CommonMarkerProps,
-    CommonMaskProps {}
+    CommonMaskProps {
+      children?: React.ReactNode;
+    }
 
 // Element props
 export interface CircleProps extends CommonPathProps {
@@ -245,6 +247,7 @@ export type Circle = React.ComponentClass<CircleProps>;
 
 export interface ClipPathProps {
   id?: string;
+  children?: React.ReactNode;
 }
 export const ClipPath: React.ComponentClass<ClipPathProps>;
 export type ClipPath = React.ComponentClass<ClipPathProps>;
@@ -314,6 +317,7 @@ export interface LinearGradientProps {
   gradientUnits?: Units;
   gradientTransform?: ColumnMajorTransformMatrix | string;
   id?: string;
+  children?: React.ReactNode;
 }
 export const LinearGradient: React.ComponentClass<LinearGradientProps>;
 export type LinearGradient = React.ComponentClass<LinearGradientProps>;
@@ -365,6 +369,7 @@ export interface RadialGradientProps {
   gradientUnits?: Units;
   gradientTransform?: ColumnMajorTransformMatrix | string;
   id?: string;
+  children?: React.ReactNode;
 }
 export const RadialGradient: React.ComponentClass<RadialGradientProps>;
 export type RadialGradient = React.ComponentClass<RadialGradientProps>;
@@ -408,6 +413,7 @@ export interface SymbolProps {
   viewBox?: string;
   preserveAspectRatio?: string;
   opacity?: NumberProp;
+  children?: React.ReactNode;
 }
 export const Symbol: React.ComponentClass<SymbolProps>;
 export type Symbol = React.ComponentClass<SymbolProps>;
@@ -441,6 +447,7 @@ export interface TextProps extends TextSpecificProps {
   rotate?: NumberArray;
   opacity?: NumberProp;
   inlineSize?: NumberProp;
+  children?: React.ReactNode;
 }
 export const Text: React.ComponentClass<TextProps>;
 export type Text = React.ComponentClass<TextProps>;
