diff --git a/node_modules/react-native-svg/apple/Elements/RNSVGImage.mm b/node_modules/react-native-svg/apple/Elements/RNSVGImage.mm
index e6532c9..0b895fc 100644
--- a/node_modules/react-native-svg/apple/Elements/RNSVGImage.mm
+++ b/node_modules/react-native-svg/apple/Elements/RNSVGImage.mm
@@ -122,14 +122,14 @@ using namespace facebook::react;
 {
   if (_state) {
     auto &observerCoordinator = _state->getData().getImageRequest().getObserverCoordinator();
-    observerCoordinator.removeObserver(*_imageResponseObserverProxy);
+    observerCoordinator.removeObserver(_imageResponseObserverProxy);
   }
 
   _state = state;
 
   if (_state) {
     auto &observerCoordinator = _state->getData().getImageRequest().getObserverCoordinator();
-    observerCoordinator.addObserver(*_imageResponseObserverProxy);
+    observerCoordinator.addObserver(_imageResponseObserverProxy);
   }
 }
 
