diff --git a/node_modules/react-native-view-shot/ios/RNViewShot.mm b/node_modules/react-native-view-shot/ios/RNViewShot.mm
index 2cf632f..c8b04e1 100644
--- a/node_modules/react-native-view-shot/ios/RNViewShot.mm
+++ b/node_modules/react-native-view-shot/ios/RNViewShot.mm
@@ -80,13 +80,12 @@ RCT_EXPORT_METHOD(captureRef:(nonnull NSNumber *)target
     UIView* rendered;
     UIScrollView* scrollView;
     if (snapshotContentContainer) {
-      if (![view isKindOfClass:[RCTScrollView class]]) {
-        reject(RCTErrorUnspecified, [NSString stringWithFormat:@"snapshotContentContainer can only be used on a RCTScrollView. instead got: %@", view], nil);
+      if (![view isKindOfClass:[UIScrollView class]]) {
+        reject(RCTErrorUnspecified, [NSString stringWithFormat:@"snapshotContentContainer can only be used on a UIScrollView. instead got: %@", view], nil);
         return;
       }
-      RCTScrollView* rctScrollView = view;
-      scrollView = rctScrollView.scrollView;
-      rendered = scrollView;
+      scrollView = (UIScrollView *)view;
+      rendered = view;
     }
     else {
       rendered = view;
@@ -197,3 +196,4 @@ RCT_EXPORT_METHOD(captureRef:(nonnull NSNumber *)target
 
 
 @end
+
