diff --git a/node_modules/react-native-webview/src/RNCWebViewNativeComponent.ts b/node_modules/react-native-webview/src/RNCWebViewNativeComponent.ts
index e2ddd36..61e7e21 100644
--- a/node_modules/react-native-webview/src/RNCWebViewNativeComponent.ts
+++ b/node_modules/react-native-webview/src/RNCWebViewNativeComponent.ts
@@ -73,13 +73,13 @@ export type WebViewNavigationEvent = Readonly<{
   canGoBack: boolean;
   canGoForward: boolean;
   lockIdentifier: Double;
-  navigationType:
-    | 'click'
-    | 'formsubmit'
-    | 'backforward'
-    | 'reload'
-    | 'formresubmit'
-    | 'other';
+  navigationType: string;
+    // | 'click'
+    // | 'formsubmit'
+    // | 'backforward'
+    // | 'reload'
+    // | 'formresubmit'
+    // | 'other';
   mainDocumentURL?: string;
 }>;
 
@@ -90,13 +90,13 @@ export type ShouldStartLoadRequestEvent = Readonly<{
   canGoBack: boolean;
   canGoForward: boolean;
   lockIdentifier: Double;
-  navigationType:
-    | 'click'
-    | 'formsubmit'
-    | 'backforward'
-    | 'reload'
-    | 'formresubmit'
-    | 'other';
+  navigationType: string;
+    // | 'click'
+    // | 'formsubmit'
+    // | 'backforward'
+    // | 'reload'
+    // | 'formresubmit'
+    // | 'other';
   mainDocumentURL?: string;
   isTopFrame: boolean;
 }>;
