diff --git a/node_modules/react-native/Libraries/AppDelegate/RCTAppDelegate.h b/node_modules/react-native/Libraries/AppDelegate/RCTAppDelegate.h
index 6f9878e..c44a0f4 100644
--- a/node_modules/react-native/Libraries/AppDelegate/RCTAppDelegate.h
+++ b/node_modules/react-native/Libraries/AppDelegate/RCTAppDelegate.h
@@ -74,6 +74,7 @@ NS_ASSUME_NONNULL_BEGIN
 @property (nonatomic, nullable) RCTSurfacePresenterBridgeAdapter *bridgeAdapter;

 - (RCTRootViewFactory *)rootViewFactory;
+- (RCTReactNativeFactory *)createReactNativeFactory;

 @end

diff --git a/node_modules/react-native/Libraries/AppDelegate/RCTAppDelegate.mm b/node_modules/react-native/Libraries/AppDelegate/RCTAppDelegate.mm
index e9321dd..3937421 100644
--- a/node_modules/react-native/Libraries/AppDelegate/RCTAppDelegate.mm
+++ b/node_modules/react-native/Libraries/AppDelegate/RCTAppDelegate.mm
@@ -44,7 +44,8 @@ using namespace facebook::react;

 - (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions
 {
-  self.reactNativeFactory = [[RCTReactNativeFactory alloc] initWithDelegate:self];
+	// TODO verify this works with no override
+  self.reactNativeFactory = [self createReactNativeFactory] ?: [[RCTReactNativeFactory alloc] initWithDelegate:self];

   if (self.automaticallyLoadReactNativeWindow) {
     [self loadReactNativeWindow:launchOptions];
diff --git a/node_modules/react-native/Libraries/AppDelegate/RCTReactNativeFactory.h b/node_modules/react-native/Libraries/AppDelegate/RCTReactNativeFactory.h
index 43a31b4..660ad01 100644
--- a/node_modules/react-native/Libraries/AppDelegate/RCTReactNativeFactory.h
+++ b/node_modules/react-native/Libraries/AppDelegate/RCTReactNativeFactory.h
@@ -42,6 +42,7 @@ NS_ASSUME_NONNULL_BEGIN
 @property (nonatomic, strong) id<RCTDependencyProvider> dependencyProvider;

 @optional
+-(RCTRootViewFactory*) createRCTRootViewFactory;
 /**
  * It creates a `RCTBridge` using a delegate and some launch options.
  * By default, it is invoked passing `self` as a delegate.
@@ -80,8 +81,12 @@ NS_ASSUME_NONNULL_BEGIN

 @end

+
+
 @interface RCTReactNativeFactory : NSObject

+//- (RCTRootViewFactory *)createRCTRootViewFactory;
+
 - (instancetype)initWithDelegate:(id<RCTReactNativeFactoryDelegate>)delegate;

 @property (nonatomic, nullable) RCTBridge *bridge;
diff --git a/node_modules/react-native/Libraries/AppDelegate/RCTReactNativeFactory.mm b/node_modules/react-native/Libraries/AppDelegate/RCTReactNativeFactory.mm
index 75c963a..6eb62a6 100644
--- a/node_modules/react-native/Libraries/AppDelegate/RCTReactNativeFactory.mm
+++ b/node_modules/react-native/Libraries/AppDelegate/RCTReactNativeFactory.mm
@@ -57,7 +57,14 @@ using namespace facebook::react;
     [RCTColorSpaceUtils applyDefaultColorSpace:[self defaultColorSpace]];
     RCTEnableTurboModule([self turboModuleEnabled]);

-    self.rootViewFactory = [self createRCTRootViewFactory];
+
+		if ([_delegate respondsToSelector:@selector(createRCTRootViewFactory)]) {
+			self.rootViewFactory = [_delegate createRCTRootViewFactory];
+		} else {
+			self.rootViewFactory = [self createRCTRootViewFactory];
+		}
+
+//    self.rootViewFactory = [self createRCTRootViewFactory]; //original

     if (newArchEnabled || fabricEnabled) {
       [RCTComponentViewFactory currentComponentViewFactory].thirdPartyFabricComponentsProvider = self;
@@ -253,9 +260,12 @@ class RCTAppDelegateBridgelessFeatureFlags : public ReactNativeFeatureFlagsDefau
   }
 };

+static BOOL overrideFF = YES; //TODO delete this!!
+
 - (void)_setUpFeatureFlags
 {
-  if ([self bridgelessEnabled]) {
+	if ([self bridgelessEnabled] && overrideFF) {
+		overrideFF = NO;
     ReactNativeFeatureFlags::override(std::make_unique<RCTAppDelegateBridgelessFeatureFlags>());
   }
 }
