diff --git a/android/app/src/main/java/com/awesomeproject/MainActivity.java b/android/app/src/main/java/com/awesomeproject/MainActivity.java
index 2a39cb8..0c6dad9 100644
--- a/android/app/src/main/java/com/awesomeproject/MainActivity.java
+++ b/android/app/src/main/java/com/awesomeproject/MainActivity.java
@@ -1,15 +1,38 @@
 package com.awesomeproject;
 
-import com.facebook.react.ReactActivity;
+import android.content.Intent;
 
-public class MainActivity extends ReactActivity {
+import com.facebook.react.ReactActivityDelegate;
+import com.facebook.react.ReactRootView;
 
+import expo.modules.devlauncher.DevLauncherController;
+import expo.modules.devmenu.react.DevMenuAwareReactActivity;
+
+public class MainActivity extends DevMenuAwareReactActivity {
+ @Override
+  public void onNewIntent(Intent intent) {
+    if (DevLauncherController.tryToHandleIntent(this, intent)) {
+      return;
+    }
+    super.onNewIntent(intent);
+  }
+
   /**
    * Returns the name of the main component registered from JavaScript. This is used to schedule
    * rendering of the component.
    */
   @Override
   protected String getMainComponentName() {
     return "MyApp";
   }
+
+  @Override
+  protected ReactActivityDelegate createReactActivityDelegate() {
+    return DevLauncherController.wrapReactActivityDelegate(this, () -> new ReactActivityDelegate(this, getMainComponentName()) {
+      @Override
+      protected ReactRootView createRootView() {
+        return new ReactRootView(MainActivity.this);
+      }
+    });
+  }
 }
diff --git a/android/app/src/main/java/com/awesomeproject/MainApplication.java b/android/app/src/main/java/com/awesomeproject/MainApplication.java
index aee0454..2b32010 100644
--- a/android/app/src/main/java/com/awesomeproject/MainApplication.java
+++ b/android/app/src/main/java/com/awesomeproject/MainApplication.java
@@ -11,6 +11,8 @@ import com.facebook.soloader.SoLoader;
 import java.lang.reflect.InvocationTargetException;
 import java.util.List;
 
+import expo.modules.devlauncher.DevLauncherController;
+
 public class MainApplication extends Application implements ReactApplication {
 
   private final ReactNativeHost mReactNativeHost =
@@ -44,6 +46,7 @@ public class MainApplication extends Application implements ReactApplication {
   public void onCreate() {
     super.onCreate();
     SoLoader.init(this, /* native exopackage */ false);
+    DevLauncherController.initialize(this, getReactNativeHost());
     initializeFlipper(this, getReactNativeHost().getReactInstanceManager());
   }
