diff --git a/android/app/src/main/java/com/expodevexample/MainActivity.java b/android/app/src/main/java/com/expodevexample/MainActivity.java
index ca9e0d1..4ee67d1 100644
--- a/android/app/src/main/java/com/expodevexample/MainActivity.java
+++ b/android/app/src/main/java/com/expodevexample/MainActivity.java
@@ -2,14 +2,18 @@ package com.expodevexample;

 import android.os.Build;
 import android.os.Bundle;
+import android.content.Intent;

 import com.facebook.react.ReactActivity;
 import com.facebook.react.ReactActivityDelegate;
 import com.facebook.react.ReactRootView;

+import expo.modules.devlauncher.DevLauncherController;
+import expo.modules.devmenu.react.DevMenuAwareReactActivity;
 import expo.modules.ReactActivityDelegateWrapper;

-public class MainActivity extends ReactActivity {
+public class MainActivity extends DevMenuAwareReactActivity {
+
   @Override
   protected void onCreate(Bundle savedInstanceState) {
     // Set the theme to AppTheme BEFORE onCreate to support
@@ -30,11 +34,23 @@ public class MainActivity extends ReactActivity {

   @Override
   protected ReactActivityDelegate createReactActivityDelegate() {
-    return new ReactActivityDelegateWrapper(this,
-      new ReactActivityDelegate(this, getMainComponentName())
+    return DevLauncherController.wrapReactActivityDelegate(
+      this,
+      () -> new ReactActivityDelegateWrapper(
+        this,
+        new ReactActivityDelegate(this, getMainComponentName())
+      )
     );
   }

+  @Override
+  public void onNewIntent(Intent intent) {
+    if (DevLauncherController.tryToHandleIntent(this, intent)) {
+      return;
+    }
+    super.onNewIntent(intent);
+  }
diff --git a/android/app/src/main/java/com/expodevexample/MainApplication.java b/android/app/src/main/java/com/expodevexample/MainApplication.java
index 78736bf..bce13b9 100644
--- a/android/app/src/main/java/com/expodevexample/MainApplication.java
+++ b/android/app/src/main/java/com/expodevexample/MainApplication.java
@@ -13,6 +13,7 @@ import com.facebook.react.ReactPackage;
 import com.facebook.soloader.SoLoader;

 import expo.modules.ApplicationLifecycleDispatcher;
+import expo.modules.devlauncher.DevLauncherController;
 import expo.modules.ReactNativeHostWrapper;

 import com.facebook.react.bridge.JSIModulePackage;
@@ -54,6 +55,7 @@ public class MainApplication extends Application implements ReactApplication {
     super.onCreate();
     SoLoader.init(this, /* native exopackage */ false);

+    DevLauncherController.initialize(this, getReactNativeHost());
     initializeFlipper(this, getReactNativeHost().getReactInstanceManager());
     ApplicationLifecycleDispatcher.onApplicationCreate(this);
   }