diff --git a/android/app/src/main/java/com/expodevexample/MainActivity.java b/android/app/src/main/java/com/expodevexample/MainActivity.java
index ca9e0d1..4ee67d1 100644
--- a/android/app/src/main/java/com/expodevexample/MainActivity.java
+++ b/android/app/src/main/java/com/expodevexample/MainActivity.java
@@ -1,16 +1,18 @@
 package com.expodevexample;
 
+import android.content.Intent;
 import android.os.Bundle;
 
-import com.facebook.react.ReactActivity;
 import com.facebook.react.ReactActivityDelegate;
 import com.facebook.react.ReactRootView;
 import com.swmansion.gesturehandler.react.RNGestureHandlerEnabledRootView;
 
+import expo.modules.devlauncher.DevLauncherController;
+import expo.modules.devmenu.react.DevMenuAwareReactActivity;
 import expo.modules.splashscreen.singletons.SplashScreen;
 import expo.modules.splashscreen.SplashScreenImageResizeMode;
 
-public class MainActivity extends ReactActivity {
+public class MainActivity extends DevMenuAwareReactActivity {
   @Override
   protected void onCreate(Bundle savedInstanceState) {
     super.onCreate(savedInstanceState);
@@ -31,11 +33,21 @@ public class MainActivity extends ReactActivity {
 
     @Override
     protected ReactActivityDelegate createReactActivityDelegate() {
-      return new ReactActivityDelegateWrapper(
-        this,
-        new ReactActivityDelegate(this, getMainComponentName()) {
-          @Override
-          protected ReactRootView createRootView() {
-            return new RNGestureHandlerEnabledRootView(MainActivity.this);
-          }
-      });
+      return DevLauncherController.wrapReactActivityDelegate(
+        this,
+        () -> new ReactActivityDelegateWrapper(
+          this,
+          new ReactActivityDelegate(this, getMainComponentName()) {
+            @Override
+            protected ReactRootView createRootView() {
+              return new RNGestureHandlerEnabledRootView(MainActivity.this);
+            }
+          }
+        )
+      );
     }
+
+    @Override
+    public void onNewIntent(Intent intent) {
+        if (DevLauncherController.tryToHandleIntent(this, intent)) {
+           return;
+        }
+        super.onNewIntent(intent);
     }
 }
diff --git a/android/app/src/main/java/com/expodevexample/MainApplication.java b/android/app/src/main/java/com/expodevexample/MainApplication.java
index 78736bf..bce13b9 100644
--- a/android/app/src/main/java/com/expodevexample/MainApplication.java
+++ b/android/app/src/main/java/com/expodevexample/MainApplication.java
@@ -19,6 +19,7 @@ import org.unimodules.adapters.react.ReactModuleRegistryProvider;
 import org.unimodules.core.interfaces.Package;
 import org.unimodules.core.interfaces.SingletonModule;
 import expo.modules.constants.ConstantsPackage;
+import expo.modules.devlauncher.DevLauncherController;
 import expo.modules.permissions.PermissionsPackage;
 import expo.modules.filesystem.FileSystemPackage;
 import expo.modules.updates.UpdatesController;
 
 public class MainApplication extends Application implements ReactApplication {
@@ -77,6 +78,7 @@ public class MainApplication extends Application implements ReactApplication {
   @Override
   public void onCreate() {
     super.onCreate();
     SoLoader.init(this, /* native exopackage */ false);

@@ -85,6 +87,7 @@ public class MainApplication extends Application implements ReactApplication {
    

+    DevLauncherController.initialize(this, getReactNativeHost());
     initializeFlipper(this, getReactNativeHost().getReactInstanceManager());
   }