diff --git a/ios/Podfile b/ios/Podfile
index 55100df..375d171 100644
--- a/ios/Podfile
+++ b/ios/Podfile
@@ -2,10 +2,14 @@ require_relative '../node_modules/react-native/scripts/react_native_pods'
 require_relative '../node_modules/react-native-unimodules/cocoapods.rb'
 require_relative '../node_modules/@react-native-community/cli-platform-ios/native_modules'
 
-platform :ios, '10.0'
+platform :ios, '12.0'
 
 target 'devclient201227' do
   use_unimodules!
+
+  pod 'expo-dev-menu', path: '../node_modules/expo-dev-menu', :configurations => :debug
+  pod 'expo-dev-launcher', path: '../node_modules/expo-dev-launcher', :configurations => :debug
+
   config = use_native_modules!
 
   use_react_native!(:path => config["reactNativePath"])
