diff --git a/android/app/src/debug/java/com/test/expoexample/ReactNativeFlipper.java b/android/app/src/debug/java/com/test/expoexample/ReactNativeFlipper.java
index 77543f3..d728f62 100644
--- a/android/app/src/debug/java/com/test/expoexample/ReactNativeFlipper.java
+++ b/android/app/src/debug/java/com/test/expoexample/ReactNativeFlipper.java
@@ -10,6 +10,7 @@ import android.content.Context;
 import com.facebook.flipper.android.AndroidFlipperClient;
 import com.facebook.flipper.android.utils.FlipperUtils;
 import com.facebook.flipper.core.FlipperClient;
+import com.facebook.flipper.core.FlipperPlugin;
 import com.facebook.flipper.plugins.crashreporter.CrashReporterPlugin;
 import com.facebook.flipper.plugins.databases.DatabasesFlipperPlugin;
 import com.facebook.flipper.plugins.fresco.FrescoFlipperPlugin;
@@ -22,11 +23,35 @@ import com.facebook.flipper.plugins.sharedpreferences.SharedPreferencesFlipperPl
 import com.facebook.react.ReactInstanceManager;
 import com.facebook.react.bridge.ReactContext;
 import com.facebook.react.modules.network.NetworkingModule;
+
+import java.util.Arrays;
+import java.util.List;
+
 import tech.bam.rnperformance.flipper.RNPerfMonitorPlugin;
 import okhttp3.OkHttpClient;
 
 public class ReactNativeFlipper {
   public static void initializeFlipper(Context context, ReactInstanceManager reactInstanceManager) {
+      if (AndroidFlipperClient.getInstanceIfInitialized() != null) {
+          final FlipperClient client = AndroidFlipperClient.getInstanceIfInitialized();
+          client.stop();
+          final List<Class> pluginClasses = Arrays.asList(
+                  InspectorFlipperPlugin.class,
+                  ReactFlipperPlugin.class,
+                  DatabasesFlipperPlugin.class,
+                  SharedPreferencesFlipperPlugin.class,
+                  CrashReporterPlugin.class,
+                  RNPerfMonitorPlugin.class,
+                  NetworkFlipperPlugin.class,
+                  FrescoFlipperPlugin.class
+          );
+          for (Class pluginClass : pluginClasses) {
+              FlipperPlugin plugin = client.getPluginByClass(pluginClass);
+              if (plugin != null) {
+                  client.removePlugin(plugin);
+              }
+          }
+      }
     if (FlipperUtils.shouldEnableFlipper(context)) {
       final FlipperClient client = AndroidFlipperClient.getInstance(context);
       client.addPlugin(new InspectorFlipperPlugin(context, DescriptorMapping.withDefaults()));