diff --git a/packages/expo-gl/android/build.gradle b/packages/expo-gl/android/build.gradle
index 32784fe8a6..bbbea44599 100644
--- a/packages/expo-gl/android/build.gradle
+++ b/packages/expo-gl/android/build.gradle
@@ -21,6 +21,10 @@ file("$REACT_NATIVE_DIR/ReactAndroid/gradle.properties").withInputStream { react
 def REACT_NATIVE_VERSION = System.getenv("REACT_NATIVE_OVERRIDE_VERSION") ?: reactProperties.getProperty("VERSION_NAME")
 def REACT_NATIVE_TARGET_VERSION = REACT_NATIVE_VERSION.split("\\.")[1].toInteger()

+REACT_NATIVE_DIR = "${rootDir}/versioned-react-native"
+REACT_NATIVE_BUILD_FROM_SOURCE = false
+RN_SO_DIR = "${buildDir}/reactandroid-{VERSIONED_ABI_NAME}-*/jni"
+
 def reactNativeArchitectures() {
     def value = project.getProperties().get("reactNativeArchitectures")
     return value ? value.split(",") : ["armeabi-v7a", "x86", "x86_64", "arm64-v8a"]
diff --git a/packages/expo-gl/android/src/main/cpp/EXGLJniApi.cpp b/packages/expo-gl/android/src/main/cpp/EXGLJniApi.cpp
index e9211d47ad..6b2c3fbb85 100644
--- a/packages/expo-gl/android/src/main/cpp/EXGLJniApi.cpp
+++ b/packages/expo-gl/android/src/main/cpp/EXGLJniApi.cpp
@@ -15,13 +15,13 @@ extern "C" {
 thread_local JNIEnv* threadLocalEnv;
 
 JNIEXPORT jint JNICALL
-Java_expo_modules_gl_cpp_EXGL_EXGLContextCreate
+Java_{VERSIONED_ABI_NAME_JNI_ESCAPED}_expo_modules_gl_cpp_EXGL_EXGLContextCreate
 (JNIEnv *env, jclass clazz) {
   return EXGLContextCreate();
 }
 
 JNIEXPORT void JNICALL
-Java_expo_modules_gl_cpp_EXGL_EXGLContextPrepare
+Java_{VERSIONED_ABI_NAME_JNI_ESCAPED}_expo_modules_gl_cpp_EXGL_EXGLContextPrepare
 (JNIEnv *env, jclass clazz, jlong jsiPtr, jint exglCtxId, jobject glContext) {
   threadLocalEnv = env;
   jclass GLContextClass = env->GetObjectClass(glContext);
@@ -35,55 +35,55 @@ Java_expo_modules_gl_cpp_EXGL_EXGLContextPrepare
 }
 
 JNIEXPORT void JNICALL
-Java_expo_modules_gl_cpp_EXGL_EXGLContextDestroy
+Java_{VERSIONED_ABI_NAME_JNI_ESCAPED}_expo_modules_gl_cpp_EXGL_EXGLContextDestroy
 (JNIEnv *env, jclass clazz, jint exglCtxId) {
   EXGLContextDestroy(exglCtxId);
 }
 
 JNIEXPORT void JNICALL
-Java_expo_modules_gl_cpp_EXGL_EXGLContextFlush
+Java_{VERSIONED_ABI_NAME_JNI_ESCAPED}_expo_modules_gl_cpp_EXGL_EXGLContextFlush
 (JNIEnv *env, jclass clazz, jint exglCtxId) {
   EXGLContextFlush(exglCtxId);
 }
 
 JNIEXPORT jint JNICALL
-Java_expo_modules_gl_cpp_EXGL_EXGLContextCreateObject
+Java_{VERSIONED_ABI_NAME_JNI_ESCAPED}_expo_modules_gl_cpp_EXGL_EXGLContextCreateObject
 (JNIEnv *env, jclass clazz, jint exglCtxId) {
   return EXGLContextCreateObject(exglCtxId);
 }
 
 JNIEXPORT void JNICALL
-Java_expo_modules_gl_cpp_EXGL_EXGLContextDestroyObject
+Java_{VERSIONED_ABI_NAME_JNI_ESCAPED}_expo_modules_gl_cpp_EXGL_EXGLContextDestroyObject
 (JNIEnv *env, jclass clazz, jint exglCtxId, jint exglObjId) {
   EXGLContextDestroyObject(exglCtxId, exglObjId);
 }
 
 JNIEXPORT void JNICALL
-Java_expo_modules_gl_cpp_EXGL_EXGLContextMapObject
+Java_{VERSIONED_ABI_NAME_JNI_ESCAPED}_expo_modules_gl_cpp_EXGL_EXGLContextMapObject
 (JNIEnv *env, jclass clazz, jint exglCtxId, jint exglObjId, jint glObj) {
   EXGLContextMapObject(exglCtxId, exglObjId, glObj);
 }
 
 JNIEXPORT jint JNICALL
-Java_expo_modules_gl_cpp_EXGL_EXGLContextGetObject
+Java_{VERSIONED_ABI_NAME_JNI_ESCAPED}_expo_modules_gl_cpp_EXGL_EXGLContextGetObject
 (JNIEnv *env, jclass clazz, jint exglCtxId, jint exglObjId) {
   return EXGLContextGetObject(exglCtxId, exglObjId);
 }
 
 JNIEXPORT void JNICALL
-Java_expo_modules_gl_cpp_EXGL_EXGLRegisterThread
+Java_{VERSIONED_ABI_NAME_JNI_ESCAPED}_expo_modules_gl_cpp_EXGL_EXGLRegisterThread
 (JNIEnv *env, jclass clazz) {
   threadLocalEnv = env;
 }
 
 JNIEXPORT bool JNICALL
-Java_expo_modules_gl_cpp_EXGL_EXGLContextNeedsRedraw
+Java_{VERSIONED_ABI_NAME_JNI_ESCAPED}_expo_modules_gl_cpp_EXGL_EXGLContextNeedsRedraw
 (JNIEnv *env, jclass clazz, jint exglCtxId) {
   return EXGLContextNeedsRedraw(exglCtxId);
 }
 
 JNIEXPORT void JNICALL
-Java_expo_modules_gl_cpp_EXGL_EXGLContextDrawEnded
+Java_{VERSIONED_ABI_NAME_JNI_ESCAPED}_expo_modules_gl_cpp_EXGL_EXGLContextDrawEnded
 (JNIEnv *env, jclass clazz, jint exglCtxId) {
   EXGLContextDrawEnded(exglCtxId);
 }
