diff --git a/packages/expo-av/android/build.gradle b/packages/expo-av/android/build.gradle
index 23806e7b3c..780fab15bb 100644
--- a/packages/expo-av/android/build.gradle
+++ b/packages/expo-av/android/build.gradle
@@ -12,6 +12,9 @@ def REACT_NATIVE_DIR = REACT_NATIVE_BUILD_FROM_SOURCE
   ? findProject(":ReactAndroid").getProjectDir().parent
   : new File(["node", "--print", "require.resolve('react-native/package.json')"].execute(null, rootDir).text.trim()).parent
 
+REACT_NATIVE_BUILD_FROM_SOURCE = false
+REACT_NATIVE_DIR = "${rootDir}/versioned-react-native/packages/react-native"
+
 def reactNativeArchitectures() {
     def value = project.getProperties().get("reactNativeArchitectures")
     return value ? value.split(",") : ["armeabi-v7a", "x86", "x86_64", "arm64-v8a"]
