diff --git a/ios/DevClientMinimalExample/AppDelegate.h b/ios/DevClientMinimalExample/AppDelegate.h
index 1a1a48f..77cbbeb 100644
--- a/ios/DevClientMinimalExample/AppDelegate.h
+++ b/ios/DevClientMinimalExample/AppDelegate.h
@@ -5,6 +5,8 @@
 
 #import <UMCore/UMAppDelegateWrapper.h>
 
-@interface AppDelegate : UMAppDelegateWrapper <RCTBridgeDelegate, EXUpdatesAppControllerDelegate>
+#import <EXDevelopmentClientController.h>
+
+@interface AppDelegate : UMAppDelegateWrapper <RCTBridgeDelegate, EXUpdatesAppControllerDelegate, EXDevelopmentClientControllerDelegate>
 
 @end
diff --git a/ios/DevClientMinimalExample/AppDelegate.m b/ios/DevClientMinimalExample/AppDelegate.m
index 90502b1..a152229 100644
--- a/ios/DevClientMinimalExample/AppDelegate.m
+++ b/ios/DevClientMinimalExample/AppDelegate.m
@@ -10,6 +10,8 @@
 #import <EXSplashScreen/EXSplashScreenService.h>
 #import <UMCore/UMModuleRegistryProvider.h>
 
+@import EXDevMenu;
+
 #ifdef FB_SONARKIT_ENABLED
 #import <FlipperKit/FlipperClient.h>
 #import <FlipperKitLayoutPlugin/FlipperKitLayoutPlugin.h>
@@ -48,7 +50,7 @@ static void InitializeFlipper(UIApplication *application) {
   self.launchOptions = launchOptions;
   self.window = [[UIWindow alloc] initWithFrame:[UIScreen mainScreen].bounds];
   #ifdef DEBUG
-    [self initializeReactNativeApp];
+    [EXDevelopmentClientController.sharedInstance startWithWindow:self.window delegate:self launchOptions:self.launchOptions];
   #else
     EXUpdatesAppController *controller = [EXUpdatesAppController sharedInstance];
     controller.delegate = self;
@@ -71,6 +73,7 @@ static void InitializeFlipper(UIApplication *application) {
   self.window.rootViewController = rootViewController;
   [self.window makeKeyAndVisible];
 
+  [DevMenuManager configureWithBridge:bridge];
   return bridge;
  }
 
@@ -83,7 +86,7 @@ static void InitializeFlipper(UIApplication *application) {
 
 - (NSURL *)sourceURLForBridge:(RCTBridge *)bridge {
  #ifdef DEBUG
-  return [[RCTBundleURLProvider sharedSettings] jsBundleURLForBundleRoot:@"index" fallbackResource:nil];
+  return EXDevelopmentClientController.sharedInstance.sourceUrl;
  #else
   return [[EXUpdatesAppController sharedInstance] launchAssetUrl];
  #endif
@@ -95,4 +98,8 @@ static void InitializeFlipper(UIApplication *application) {
   [splashScreenService showSplashScreenFor:self.window.rootViewController];
 }
 
+- (void)developmentClientController:(EXDevelopmentClientController *)developmentClientController didStartWithSuccess:(BOOL)success {
+  developmentClientController.appBridge = [self initializeReactNativeApp];
+}
+
 @end
