diff --git a/android/app/build.gradle b/android/app/build.gradle
index 0d8de38..c0951ea 100644
--- a/android/app/build.gradle
+++ b/android/app/build.gradle
@@ -1,4 +1,5 @@
 apply plugin: "com.android.application"
 
 import com.android.build.OutputFile
 
@@ -186,7 +187,8 @@ dependencies {
     implementation "com.facebook.react:react-native:+"  // From node_modules
 
     implementation "androidx.swiperefreshlayout:swiperefreshlayout:1.0.0"
     debugImplementation("com.facebook.flipper:flipper:${FLIPPER_VERSION}") {
       exclude group:'com.facebook.fbjni'
     }
@@ -207,6 +209,9 @@ dependencies {
     } else {
         implementation jscFlavor
     }
+
+    implementation project(":expo-dev-menu-interface")
+    implementation project(":expo-dev-menu")
 }
 
 // Run this once to be able to run the application with BUCK
diff --git a/android/app/src/main/java/com/dc0925/MainActivity.java b/android/app/src/main/java/com/dc0925/MainActivity.java
index 1804f20..a73c2cb 100644
--- a/android/app/src/main/java/com/dc0925/MainActivity.java
+++ b/android/app/src/main/java/com/dc0925/MainActivity.java
@@ -1,8 +1,9 @@
 package com.dc0925;
 
-import com.facebook.react.ReactActivity;
+import expo.modules.devmenu.react.DevMenuAwareReactActivity;
 
+//
+//  Use a custom activity?  You can extend your own by following the instructions at https://github.com/expo/expo/tree/master/packages/expo-dev-menu
+//
-public class MainActivity extends ReactActivity {
+public class MainActivity extends DevMenuAwareReactActivity {
 
   /**
    * Returns the name of the main component registered from JavaScript. This is used to schedule
diff --git a/android/build.gradle b/android/build.gradle
index ed5a568..0547dbf 100644
--- a/android/build.gradle
+++ b/android/build.gradle
@@ -3,7 +3,7 @@
 buildscript {
     ext {
         buildToolsVersion = "29.0.2"
         // Make sure you upgraded the minSdkVersion 
         minSdkVersion = 21
         compileSdkVersion = 29
         targetSdkVersion = 29
     }
diff --git a/android/settings.gradle b/android/settings.gradle
index e84bb50..7e3fa13 100644
--- a/android/settings.gradle
+++ b/android/settings.gradle
@@ -1,4 +1,9 @@
 rootProject.name = 'dc0925'
 apply from: '../node_modules/react-native-unimodules/gradle.groovy'; includeUnimodulesProjects()
 apply from: file("../node_modules/@react-native-community/cli-platform-android/native_modules.gradle"); applyNativeModulesSettingsGradle(settings)
+include(":expo-dev-menu")
+project(":expo-dev-menu").projectDir = new File("../node_modules/expo-dev-menu/android")
+
+include(":expo-dev-menu-interface")
+project(":expo-dev-menu-interface").projectDir = new File("../node_modules/expo-dev-menu-interface/android")
 include ':app'
