diff --git a/android/app/build.gradle b/android/app/build.gradle
index 92f2403..ae13221 100644
--- a/android/app/build.gradle
+++ b/android/app/build.gradle
@@ -199,6 +199,8 @@ dependencies {
     }
     addUnimodulesDependencies()
 
+    implementation project(":expo-dev-menu")
+
     if (enableHermes) {
         def hermesPath = "../../node_modules/hermes-engine/android/";
         debugImplementation files(hermesPath + "hermes-debug.aar")
diff --git a/android/settings.gradle b/android/settings.gradle
index 83c7f0f..8a9ef36 100644
--- a/android/settings.gradle
+++ b/android/settings.gradle
@@ -6,4 +6,7 @@ includeUnimodulesProjects()
 apply from: file("../node_modules/@react-native-community/cli-platform-android/native_modules.gradle");
 applyNativeModulesSettingsGradle(settings)
 
+include(":expo-dev-menu")
+project(":expo-dev-menu").projectDir = new File("../node_modules/expo-dev-menu/android")
+
 include ':app'
