diff --git a/android/app/src/main/java/com/devclientminimalexample/MainActivity.java b/android/app/src/main/java/com/devclientminimalexample/MainActivity.java
index 4fc434b..e5923bc 100644
--- a/android/app/src/main/java/com/devclientminimalexample/MainActivity.java
+++ b/android/app/src/main/java/com/devclientminimalexample/MainActivity.java
@@ -7,10 +7,13 @@ import com.facebook.react.ReactActivityDelegate;
 import com.facebook.react.ReactRootView;
 import com.swmansion.gesturehandler.react.RNGestureHandlerEnabledRootView;
 
+import expo.modules.developmentclient.DevelopmentClientController;
 import expo.modules.splashscreen.singletons.SplashScreen;
 import expo.modules.splashscreen.SplashScreenImageResizeMode;
 
-public class MainActivity extends ReactActivity {
+import expo.modules.devmenu.react.DevMenuAwareReactActivity;
+
+public class MainActivity extends DevMenuAwareReactActivity {
   @Override
   protected void onCreate(Bundle savedInstanceState) {
     super.onCreate(savedInstanceState);
@@ -31,11 +34,16 @@ public class MainActivity extends ReactActivity {
 
     @Override
     protected ReactActivityDelegate createReactActivityDelegate() {
-        return new ReactActivityDelegate(this, getMainComponentName()) {
+        ReactActivityDelegate delegate =  new ReactActivityDelegate(this, getMainComponentName()) {
             @Override
             protected ReactRootView createRootView() {
                 return new RNGestureHandlerEnabledRootView(MainActivity.this);
             }
         };
+        if (BuildConfig.DEBUG) {
+          return DevelopmentClientController.wrapReactActivityDelegate(this, () -> delegate);
+        } else {
+          return delegate;
+        }
     }
 }
diff --git a/android/app/src/main/java/com/devclientminimalexample/MainApplication.java b/android/app/src/main/java/com/devclientminimalexample/MainApplication.java
index 2f3748d..abda6b0 100644
--- a/android/app/src/main/java/com/devclientminimalexample/MainApplication.java
+++ b/android/app/src/main/java/com/devclientminimalexample/MainApplication.java
@@ -19,6 +19,7 @@ import org.unimodules.adapters.react.ReactModuleRegistryProvider;
 import org.unimodules.core.interfaces.Package;
 import org.unimodules.core.interfaces.SingletonModule;
 import expo.modules.constants.ConstantsPackage;
+import expo.modules.developmentclient.DevelopmentClientController;
 import expo.modules.permissions.PermissionsPackage;
 import expo.modules.filesystem.FileSystemPackage;
 import expo.modules.updates.UpdatesController;
@@ -80,6 +81,10 @@ public class MainApplication extends Application implements ReactApplication {
     super.onCreate();
     SoLoader.init(this, /* native exopackage */ false);
 
+    if (BuildConfig.DEBUG) {
+      DevelopmentClientController.initialize(this, mReactNativeHost);
+    }
+
     if (!BuildConfig.DEBUG) {
       UpdatesController.initialize(this);
     }
