diff --git a/ios/Podfile b/ios/Podfile
index 3c8b2f5..3f3cab1 100644
--- a/ios/Podfile
+++ b/ios/Podfile
@@ -7,6 +7,8 @@ platform :ios, '10.0'
 target 'dc0925' do
   use_unimodules!
   config = use_native_modules!
+  pod 'EXDevMenuInterface', path: '../node_modules/expo-dev-menu-interface'
+  pod 'EXDevMenu', path: '../node_modules/expo-dev-menu', :configurations => :debug
 
   use_react_native!(:path => config["reactNativePath"])
 
diff --git a/ios/dc0925/AppDelegate.m b/ios/dc0925/AppDelegate.m
index 6be80ef..bc618b0 100644
--- a/ios/dc0925/AppDelegate.m
+++ b/ios/dc0925/AppDelegate.m
@@ -9,6 +9,10 @@
 #import <UMReactNativeAdapter/UMNativeModulesProxy.h>
 #import <UMReactNativeAdapter/UMModuleRegistryAdapter.h>
 
+#if __has_include(<EXDevMenu/EXDevMenu-umbrella.h>)
+@import EXDevMenu;
+#endif
+
 #ifdef FB_SONARKIT_ENABLED
 #import <FlipperKit/FlipperClient.h>
 #import <FlipperKitLayoutPlugin/FlipperKitLayoutPlugin.h>
@@ -50,6 +54,9 @@ static void InitializeFlipper(UIApplication *application) {
   RCTRootView *rootView = [[RCTRootView alloc] initWithBridge:bridge
                                                    moduleName:@"dc0925"
                                             initialProperties:nil];
+  #if __has_include(<EXDevMenu/EXDevMenu-umbrella.h>)
+  [DevMenuManager configureWithBridge:bridge];
+  #endif
 
   rootView.backgroundColor = [[UIColor alloc] initWithRed:1.0f green:1.0f blue:1.0f alpha:1];
 
