diff --git a/android/app/build.gradle b/android/app/build.gradle
index c99d3e9..990fddf 100644
--- a/android/app/build.gradle
+++ b/android/app/build.gradle
@@ -266,6 +266,21 @@ android {

         }
     }
+
+    packagingOptions {
+        pickFirsts = [
+            "lib/**/libc++_shared.so",
+            "lib/**/libreactnativejni.so",
+            "lib/**/libreact_nativemodule_core.so",
+            "lib/**/libglog.so",
+            "lib/**/libjscexecutor.so",
+            "lib/**/libfbjni.so",
+            "lib/**/libfolly_json.so",
+            "lib/**/libfolly_runtime.so",
+            "lib/**/libhermes.so",
+            "lib/**/libjsi.so",
+        ]
+    }
 }

 // Apply static values from `gradle.properties` to the `android.packagingOptions`
