diff --git a/e2e/homeScreen.test.js b/e2e/homeScreen.test.js
index e28acfa..c65e90c 100644
--- a/e2e/homeScreen.test.js
+++ b/e2e/homeScreen.test.js
@@ -8,7 +8,7 @@ describe('Home screen', () => {
   });

   it('"Click me" button should be visible', async () => {
-    await expect(element(by.id('click-me-button'))).toBeVisible();
+    await expect(element(by.id('click-me-button'))).not.toBeVisible();
   });

   it('shows "Hi!" after tapping "Click me"', async () => {
