diff --git a/android/app/build.gradle b/android/app/build.gradle
index 2465ebf..a9e2343 100644
--- a/android/app/build.gradle
+++ b/android/app/build.gradle
@@ -49,6 +49,11 @@ react {
     //
     //   The list of flags to pass to the Hermes compiler. By default is "-O", "-output-source-map"
     // hermesFlags = ["-O", "-output-source-map"]
+    // Override `hermesEnabled` by `expo.jsEngine`
+    ext {
+    hermesEnabled = (findProperty('expo.jsEngine') ?: "hermes") == "hermes"
+    }
+
     entryFile = file(["node", "-e", "require('expo/scripts/resolveAppEntry')", rootDir.getAbsoluteFile().getParentFile().getAbsolutePath(), "android", "absolute"].execute(null, rootDir).text.trim())
     cliFile = new File(["node", "--print", "require.resolve('@expo/cli')"].execute(null, rootDir).text.trim())
     bundleCommand = "export:embed"