diff --git a/ios/MyApp/AppDelegate.h b/ios/MyApp/AppDelegate.h
index ef1de86..d3d75b0 100644
--- a/ios/MyApp/AppDelegate.h
+++ b/ios/MyApp/AppDelegate.h
@@ -1,6 +1,7 @@
 #import <RCTAppDelegate.h>
+#import <Expo/Expo.h>
 #import <UIKit/UIKit.h>

-@interface AppDelegate : RCTAppDelegate
+@interface AppDelegate : EXAppDelegateWrapper

 @end
diff --git a/ios/Podfile b/ios/Podfile
index 99b1b8d..71524e2 100644
--- a/ios/Podfile
+++ b/ios/Podfile
@@ -1,7 +1,8 @@
+require File.join(File.dirname(`node --print "require.resolve('expo/package.json')"`), "scripts/autolinking")
 require_relative '../node_modules/react-native/scripts/react_native_pods'
 require_relative '../node_modules/@react-native-community/cli-platform-ios/native_modules'

-platform :ios, min_ios_version_supported
+platform :ios, '13.0'
 prepare_react_native_project!

 # If you are using a `react-native-flipper` your iOS build will fail when `NO_FLIPPER=1` is set.
@@ -22,6 +23,14 @@ if linkage != nil
 end

 target 'MyApp' do
+  use_expo_modules!
+  post_integrate do |installer|
+    begin
+      expo_patch_react_imports!(installer)
+    rescue => e
+      Pod::UI.warn e
+    end
+  end
   config = use_native_modules!
