diff --git a/apps/bare-update/ios/bareupdate/AppDelegate.h b/apps/bare-update/ios/bareupdate/AppDelegate.h
index e39065416f..7a2f733b9d 100644
--- a/apps/bare-update/ios/bareupdate/AppDelegate.h
+++ b/apps/bare-update/ios/bareupdate/AppDelegate.h
@@ -1,9 +1,10 @@
 #import <Foundation/Foundation.h>
+#import <EXUpdates/EXUpdatesAppController.h>
 #import <React/RCTBridgeDelegate.h>
 #import <UIKit/UIKit.h>
 
 #import <ExpoModulesCore/EXAppDelegateWrapper.h>
 
-@interface AppDelegate : EXAppDelegateWrapper <RCTBridgeDelegate>
+@interface AppDelegate : EXAppDelegateWrapper <RCTBridgeDelegate, EXUpdatesAppControllerDelegate>
 
 @end
diff --git a/apps/bare-update/ios/bareupdate/AppDelegate.m b/apps/bare-update/ios/bareupdate/AppDelegate.m
index e2bd877b92..b8839389ac 100644
--- a/apps/bare-update/ios/bareupdate/AppDelegate.m
+++ b/apps/bare-update/ios/bareupdate/AppDelegate.m
@@ -28,6 +28,12 @@ static void InitializeFlipper(UIApplication *application) {
 }
 #endif
 
+@interface AppDelegate () <RCTBridgeDelegate>
+
+@property (nonatomic, strong) NSDictionary *launchOptions;
+
+@end
+
 @implementation AppDelegate
 
 - (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions
@@ -36,7 +42,24 @@ - (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(
   InitializeFlipper(application);
 #endif
   
-  RCTBridge *bridge = [[RCTBridge alloc] initWithDelegate:self launchOptions:launchOptions];
+  self.launchOptions = launchOptions;
+  self.window = [[UIWindow alloc] initWithFrame:[UIScreen mainScreen].bounds];
+  #ifdef DEBUG
+    [self initializeReactNativeApp];
+  #else
+    EXUpdatesAppController *controller = [EXUpdatesAppController sharedInstance];
+    controller.delegate = self;
+    [controller startAndShowLaunchScreen:self.window];
+  #endif
+
+  [super application:application didFinishLaunchingWithOptions:launchOptions];
+
+  return YES;
+}
+
+- (RCTBridge *)initializeReactNativeApp
+{
+  RCTBridge *bridge = [[RCTBridge alloc] initWithDelegate:self launchOptions:self.launchOptions];
   RCTRootView *rootView = [[RCTRootView alloc] initWithBridge:bridge moduleName:@"main" initialProperties:nil];
   id rootViewBackgroundColor = [[NSBundle mainBundle] objectForInfoDictionaryKey:@"RCTRootViewBackgroundColor"];
   if (rootViewBackgroundColor != nil) {
@@ -45,15 +68,13 @@ - (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(
     rootView.backgroundColor = [UIColor whiteColor];
   }
 
-  self.window = [[UIWindow alloc] initWithFrame:[UIScreen mainScreen].bounds];
+
   UIViewController *rootViewController = [UIViewController new];
   rootViewController.view = rootView;
   self.window.rootViewController = rootViewController;
   [self.window makeKeyAndVisible];
 
-  [super application:application didFinishLaunchingWithOptions:launchOptions];
-
-  return YES;
+  return bridge;
  }
 
 - (NSArray<id<RCTBridgeModule>> *)extraModulesForBridge:(RCTBridge *)bridge
@@ -70,10 +91,14 @@ - (NSURL *)sourceURLForBridge:(RCTBridge *)bridge {
  #ifdef DEBUG
   return [[RCTBundleURLProvider sharedSettings] jsBundleURLForBundleRoot:@"index" fallbackResource:nil];
  #else
-  return [[NSBundle mainBundle] URLForResource:@"main" withExtension:@"jsbundle"];
+  return [[EXUpdatesAppController sharedInstance] launchAssetUrl];
  #endif
 }
 
+- (void)appController:(EXUpdatesAppController *)appController didStartWithSuccess:(BOOL)success {
+  appController.bridge = [self initializeReactNativeApp];
+}
+
 // Linking API
 - (BOOL)application:(UIApplication *)application openURL:(NSURL *)url options:(NSDictionary<UIApplicationOpenURLOptionsKey,id> *)options {
   return [RCTLinkingManager application:application openURL:url options:options];
diff --git a/apps/bare-update/ios/bareupdate/Supporting/Expo.plist b/apps/bare-update/ios/bareupdate/Supporting/Expo.plist
index 471ca3dde5..2abdd7a775 100644
--- a/apps/bare-update/ios/bareupdate/Supporting/Expo.plist
+++ b/apps/bare-update/ios/bareupdate/Supporting/Expo.plist
@@ -6,5 +6,7 @@
     <string>42.0.0</string>
     <key>EXUpdatesURL</key>
     <string>https://exp.host/@my-expo-username/my-app</string>
+    <key>EXUpdatesAutoSetup</key>
+    <false/>
   </dict>
 </plist>
