diff --git a/app.json b/app.json
index 1c3dccb..f281582 100644
--- a/app.json
+++ b/app.json
@@ -1,4 +1,9 @@
 {
   "name": "MyApp",
-  "displayName": "MyApp"
+  "displayName": "MyApp",
+  "expo": {
+    "name": "MyApp",
+    "slug": "my-app",
+    "sdkVersion": "38.0.0"
+  }
 }
\ No newline at end of file
diff --git a/index.js b/index.js
index a850d03..9d497de 100644
--- a/index.js
+++ b/index.js
@@ -1,9 +1,10 @@
 /**
  * @format
  */
 
+import 'expo-asset';
 import {AppRegistry} from 'react-native';
 import App from './App';
 import {name as appName} from './app.json';
 
 AppRegistry.registerComponent(appName, () => App);
diff --git a/metro.config.js b/metro.config.js
index 13a9642..20ef36e 100644
--- a/metro.config.js
+++ b/metro.config.js
@@ -5,10 +5,11 @@
  * @format
  */
 
 module.exports = {
   transformer: {
+    assetPlugins: ['expo-asset/tools/hashAssetFiles'],
     getTransformOptions: async () => ({
       transform: {
         experimentalImportSupport: false,
         inlineRequires: false,
       },