diff --git a/ios/myapp/AppDelegate.h b/ios/myapp/AppDelegate.h
index 5d28082..a7ebb51 100644
--- a/ios/myapp/AppDelegate.h
+++ b/ios/myapp/AppDelegate.h
@@ -1,6 +1,7 @@
 #import <RCTAppDelegate.h>
+#import <Expo/Expo.h>
 #import <UIKit/UIKit.h>

-@interface AppDelegate : RCTAppDelegate
+@interface AppDelegate : EXAppDelegateWrapper

 @end
diff --git a/ios/Podfile b/ios/Podfile
index 6bb6b6f..8c3834c 100644
--- a/ios/Podfile
+++ b/ios/Podfile
@@ -1,3 +1,4 @@
+require File.join(File.dirname(`node --print "require.resolve('expo/package.json')"`), "scripts/autolinking")
 # Resolve react_native_pods.rb with node to allow for hoisting
 require Pod::Executable.execute_command('node', ['-p',
   'require.resolve(
@@ -15,6 +16,14 @@ if linkage != nil
 end

 target 'myapp' do
+  use_expo_modules!
+  post_integrate do |installer|
+    begin
+      expo_patch_react_imports!(installer)
+    rescue => e
+      Pod::UI.warn e
+    end
+  end
   config = use_native_modules!

   use_react_native!(
