diff --git a/packages/expo-brownfield/plugin/templates/android/BrownfieldActivity.kt b/packages/expo-brownfield/plugin/templates/android/BrownfieldActivity.kt
--- a/packages/expo-brownfield/plugin/templates/android/BrownfieldActivity.kt
+++ b/packages/expo-brownfield/plugin/templates/android/BrownfieldActivity.kt
@@ -4,10 +4,12 @@
 import android.app.Application
 import android.content.res.Configuration
 import android.os.Bundle
+import android.view.KeyEvent
 import androidx.appcompat.app.AppCompatActivity
 import com.facebook.react.ReactPackage
 import com.facebook.react.modules.core.DefaultHardwareBackBtnHandler
 import expo.modules.ApplicationLifecycleDispatcher
+import expo.modules.devmenu.api.DevMenuApi
 
 object BrownfieldLifecycleDispatcher {
   @JvmStatic
@@ -47,4 +49,15 @@
   override fun invokeDefaultOnBackPressed() {
     finish()
   }
+
+  override fun onKeyUp(keyCode: Int, event: KeyEvent): Boolean {
+    if (BuildConfig.DEBUG) {
+      val fragment by DevMenuApi.fragment { this }
+      if (fragment?.onKeyUp(keyCode, event) == true) {
+        return true
+      }
+    }
+
+    return super.onKeyUp(keyCode, event)
+  }
 }
