diff --git a/packages/expo-brownfield/plugin/templates/android/BrownfieldActivity.kt b/packages/expo-brownfield/plugin/templates/android/BrownfieldActivity.kt
index dda8db9d25..ee263e78e2 100644
--- a/packages/expo-brownfield/plugin/templates/android/BrownfieldActivity.kt
+++ b/packages/expo-brownfield/plugin/templates/android/BrownfieldActivity.kt
@@ -4,6 +4,8 @@ import android.app.Application
 import android.content.res.Configuration
 import androidx.appcompat.app.AppCompatActivity
 import expo.modules.ApplicationLifecycleDispatcher
+import android.view.KeyEvent
+import expo.modules.devmenu.api.DevMenuApi
 
 object BrownfieldLifecycleDispatcher {
   fun onApplicationCreate(application: Application) {
@@ -20,4 +22,15 @@ open class BrownfieldActivity : AppCompatActivity() {
     super.onConfigurationChanged(newConfig)
     BrownfieldLifecycleDispatcher.onConfigurationChanged(this.application, newConfig)
   }
+
+  override fun onKeyUp(keyCode: Int, event: KeyEvent): Boolean {
+    if (BuildConfig.DEBUG) {
+      val fragment by DevMenuApi.fragment { this }
+      if (fragment?.onKeyUp(keyCode, event) == true) {
+        return true
+      }
+    }
+
+    return super.onKeyUp(keyCode, event)
+  }
 }
