diff --git a/packages/expo-brownfield/plugin/templates/android/build.gradle.kts b/packages/expo-brownfield/plugin/templates/android/build.gradle.kts
index a22a00aaab..f2b6d065f9 100644
--- a/packages/expo-brownfield/plugin/templates/android/build.gradle.kts
+++ b/packages/expo-brownfield/plugin/templates/android/build.gradle.kts
@@ -51,8 +51,22 @@ android {
   kotlinOptions { jvmTarget = "17" }
 }
 
+val configureDevMenuInRelease = 
+  findProperty("expo.devmenu.configureInRelease")?.toString() == "true"
+
 dependencies {
+  val debugOnly: (Any) -> Unit = { notation ->
+    debugImplementation(notation)
+    if (configureDevMenuInRelease) {
+      releaseImplementation(notation)
+    }
+  }
+
   api("com.facebook.react:react-android")
   api("com.facebook.hermes:hermes-android")
   compileOnly("androidx.fragment:fragment-ktx:1.6.1")
+
+  debugOnly(project(":expo-dev-menu"))
+  debugOnly(project(":expo-manifests"))
+  debugOnly("com.squareup.okhttp3:okhttp:4.9.2")
 }
