diff --git a/templates/expo-template-bare-minimum/android/app/src/main/java/com/helloworld/MainApplication.kt b/templates/expo-template-bare-minimum/android/app/src/main/java/com/helloworld/MainApplication.kt
index dd90f54e04..8258c40ed4 100644
--- a/templates/expo-template-bare-minimum/android/app/src/main/java/com/helloworld/MainApplication.kt
+++ b/templates/expo-template-bare-minimum/android/app/src/main/java/com/helloworld/MainApplication.kt
@@ -52,7 +52,9 @@ class MainApplication : Application(), ReactApplication {
       // If you opted-in for the New Architecture, we load the native entry point for this app.
       load()
     }
-    ReactNativeFlipper.initializeFlipper(this, reactNativeHost.reactInstanceManager)
+    if (BuildConfig.DEBUG) {
+      ReactNativeFlipper.initializeFlipper(this, reactNativeHost.reactInstanceManager)
+    }
     ApplicationLifecycleDispatcher.onApplicationCreate(this)
   }
 
diff --git a/templates/expo-template-bare-minimum/package.json b/templates/expo-template-bare-minimum/package.json
index df471f5854..a0b1b85c0a 100644
--- a/templates/expo-template-bare-minimum/package.json
+++ b/templates/expo-template-bare-minimum/package.json
@@ -1,7 +1,7 @@
 {
   "name": "expo-template-bare-minimum",
   "description": "This bare project template includes a minimal setup for using unimodules with React Native.",
-  "version": "50.0.21",
+  "version": "50.0.17",
   "main": "index.js",
   "scripts": {
     "start": "expo start --dev-client",
@@ -10,8 +10,8 @@
     "web": "expo start --web"
   },
   "dependencies": {
-    "expo": "~50.0.0-preview.7",
-    "expo-status-bar": "~1.11.1",
+    "expo": "~50.0.0-preview.4",
+    "expo-status-bar": "~1.11.0",
     "react": "18.2.0",
     "react-native": "0.73.1"
   },
