--- a/node_modules/react-native-gesture-handler/ios/RNGestureHandlerModule.mm
+++ b/node_modules/react-native-gesture-handler/ios/RNGestureHandlerModule.mm
@@ -96,7 +96,7 @@
           return jsi::Value::null();
         }
 
-        auto shadowNode = arguments[0].asObject(runtime).getHostObject<ShadowNodeWrapper>(runtime)->shadowNode;
+        auto shadowNode = arguments[0].asObject(runtime).getNativeState<ShadowNode>(runtime);
         bool isFormsStackingContext = shadowNode->getTraits().check(ShadowNodeTraits::FormsStackingContext);
 
         return jsi::Value(isFormsStackingContext);
