diff --git a/dist/build/index.js b/dist/build/index.js
index 5a19c0a61e41b76696e945e5a6b7542f0f39a782..ba219eba8fb4c4b9d24f18e289e2eb3990319c92 100644
--- a/dist/build/index.js
+++ b/dist/build/index.js
@@ -927,7 +927,7 @@ async function build(dir, experimentalAnalyze = false, reactProductionProfiling
                             (0, _trace1.traceMemoryUsage)('Finished server compilation', nextBuildSpan);
                             buildTraceContext = res.buildTraceContext;
                             durationInSeconds += res.duration;
-                            if (collectServerBuildTracesInParallel) {
+                            if (collectServerBuildTracesInParallel && config.output !== 'export') {
                                 const buildTraceWorker = new _worker.Worker(require.resolve('./collect-build-traces'), {
                                     debuggerPortOffset: -1,
                                     isolatedMemory: false,
@@ -1546,7 +1546,7 @@ async function build(dir, experimentalAnalyze = false, reactProductionProfiling
                 }
             }
             await writeFunctionsConfigManifest(distDir, functionsConfigManifest);
-            if (bundler !== _bundler.Bundler.Turbopack && !isGenerateMode && !buildTracesPromise) {
+            if (bundler !== _bundler.Bundler.Turbopack && !isGenerateMode && !buildTracesPromise && config.output !== 'export') {
                 buildTracesPromise = nextBuildSpan.traceChild('collect-build-traces').traceAsyncFn(()=>{
                     return (0, _collectbuildtraces.collectBuildTraces)({
                         dir,
diff --git a/dist/esm/build/index.js b/dist/esm/build/index.js
index 16ede18c636db7e4d6820b0a4c0f4a013d7ee90d..dbb1d1b7d3325c30789f3d4d8347c442b4078eb6 100644
--- a/dist/esm/build/index.js
+++ b/dist/esm/build/index.js
@@ -859,7 +859,7 @@ export default async function build(dir, experimentalAnalyze = false, reactProdu
                             traceMemoryUsage('Finished server compilation', nextBuildSpan);
                             buildTraceContext = res.buildTraceContext;
                             durationInSeconds += res.duration;
-                            if (collectServerBuildTracesInParallel) {
+                            if (collectServerBuildTracesInParallel && config.output !== 'export') {
                                 const buildTraceWorker = new Worker(require.resolve('./collect-build-traces'), {
                                     debuggerPortOffset: -1,
                                     isolatedMemory: false,
@@ -1478,7 +1478,7 @@ export default async function build(dir, experimentalAnalyze = false, reactProdu
                 }
             }
             await writeFunctionsConfigManifest(distDir, functionsConfigManifest);
-            if (bundler !== Bundler.Turbopack && !isGenerateMode && !buildTracesPromise) {
+            if (bundler !== Bundler.Turbopack && !isGenerateMode && !buildTracesPromise && config.output !== 'export') {
                 buildTracesPromise = nextBuildSpan.traceChild('collect-build-traces').traceAsyncFn(()=>{
                     return collectBuildTraces({
                         dir,
