diff --git a/src/private/components/virtualview/VirtualViewExperimentalNativeComponent.js b/src/private/components/virtualview/VirtualViewExperimentalNativeComponent.js
index c640b5a..fa980e8 100644
--- a/src/private/components/virtualview/VirtualViewExperimentalNativeComponent.js
+++ b/src/private/components/virtualview/VirtualViewExperimentalNativeComponent.js
@@ -18,7 +18,7 @@ import type {HostComponent} from '../../types/HostComponent';
 
 import codegenNativeComponent from '../../../../Libraries/Utilities/codegenNativeComponent';
 
-export type NativeModeChangeEvent = Readonly<{
+export type NativeModeChangeEvent = $ReadOnly<{
   /**
    * Virtualization mode of the target view.
    *
@@ -34,7 +34,7 @@ export type NativeModeChangeEvent = Readonly<{
   /**
    * Rect of the target view, relative to the nearest ancestor scroll container.
    */
-  targetRect: Readonly<{
+  targetRect: $ReadOnly<{
     x: Double,
     y: Double,
     width: Double,
@@ -51,7 +51,7 @@ export type NativeModeChangeEvent = Readonly<{
    *
    * This can be used to determine whether and how much new content to render.
    */
-  thresholdRect: Readonly<{
+  thresholdRect: $ReadOnly<{
     x: Double,
     y: Double,
     width: Double,
@@ -59,7 +59,7 @@ export type NativeModeChangeEvent = Readonly<{
   }>,
 }>;
 
-type VirtualViewExperimentalNativeProps = Readonly<{
+type VirtualViewExperimentalNativeProps = $ReadOnly<{
   ...ViewProps,
 
   /**
diff --git a/src/private/components/virtualview/VirtualViewNativeComponent.js b/src/private/components/virtualview/VirtualViewNativeComponent.js
index 4ba5b68..951a000 100644
--- a/src/private/components/virtualview/VirtualViewNativeComponent.js
+++ b/src/private/components/virtualview/VirtualViewNativeComponent.js
@@ -18,7 +18,7 @@ import type {HostComponent} from '../../types/HostComponent';
 
 import codegenNativeComponent from '../../../../Libraries/Utilities/codegenNativeComponent';
 
-export type NativeModeChangeEvent = Readonly<{
+export type NativeModeChangeEvent = $ReadOnly<{
   /**
    * Virtualization mode of the target view.
    *
@@ -34,7 +34,7 @@ export type NativeModeChangeEvent = Readonly<{
   /**
    * Rect of the target view, relative to the nearest ancestor scroll container.
    */
-  targetRect: Readonly<{
+  targetRect: $ReadOnly<{
     x: Double,
     y: Double,
     width: Double,
@@ -51,7 +51,7 @@ export type NativeModeChangeEvent = Readonly<{
    *
    * This can be used to determine whether and how much new content to render.
    */
-  thresholdRect: Readonly<{
+  thresholdRect: $ReadOnly<{
     x: Double,
     y: Double,
     width: Double,
@@ -59,7 +59,7 @@ export type NativeModeChangeEvent = Readonly<{
   }>,
 }>;
 
-type VirtualViewNativeProps = Readonly<{
+type VirtualViewNativeProps = $ReadOnly<{
   ...ViewProps,
 
   /**
diff --git a/src/private/specs_DEPRECATED/components/AndroidDrawerLayoutNativeComponent.js b/src/private/specs_DEPRECATED/components/AndroidDrawerLayoutNativeComponent.js
index 22ce1ab..d6352f7 100644
--- a/src/private/specs_DEPRECATED/components/AndroidDrawerLayoutNativeComponent.js
+++ b/src/private/specs_DEPRECATED/components/AndroidDrawerLayoutNativeComponent.js
@@ -22,15 +22,15 @@ import codegenNativeCommands from '../../../../Libraries/Utilities/codegenNative
 import codegenNativeComponent from '../../../../Libraries/Utilities/codegenNativeComponent';
 import * as React from 'react';
 
-type DrawerStateEvent = Readonly<{
+type DrawerStateEvent = $ReadOnly<{
   drawerState: Int32,
 }>;
 
-type DrawerSlideEvent = Readonly<{
+type DrawerSlideEvent = $ReadOnly<{
   offset: Float,
 }>;
 
-type AndroidDrawerLayoutNativeProps = Readonly<{
+type AndroidDrawerLayoutNativeProps = $ReadOnly<{
   ...ViewProps,
   /**
    * Determines whether the keyboard gets dismissed in response to a drag.
diff --git a/src/private/specs_DEPRECATED/components/AndroidSwipeRefreshLayoutNativeComponent.js b/src/private/specs_DEPRECATED/components/AndroidSwipeRefreshLayoutNativeComponent.js
index 35acd83..4a2d5d1 100644
--- a/src/private/specs_DEPRECATED/components/AndroidSwipeRefreshLayoutNativeComponent.js
+++ b/src/private/specs_DEPRECATED/components/AndroidSwipeRefreshLayoutNativeComponent.js
@@ -21,7 +21,7 @@ import codegenNativeCommands from '../../../../Libraries/Utilities/codegenNative
 import codegenNativeComponent from '../../../../Libraries/Utilities/codegenNativeComponent';
 import * as React from 'react';
 
-type AndroidSwipeRefreshLayoutNativeProps = Readonly<{
+type AndroidSwipeRefreshLayoutNativeProps = $ReadOnly<{
   ...ViewProps,
 
   /**
@@ -31,7 +31,7 @@ type AndroidSwipeRefreshLayoutNativeProps = Readonly<{
   /**
    * The colors (at least one) that will be used to draw the refresh indicator.
    */
-  colors?: ?ReadonlyArray<ColorValue>,
+  colors?: ?$ReadOnlyArray<ColorValue>,
   /**
    * The background color of the refresh indicator.
    */
diff --git a/src/private/specs_DEPRECATED/components/AndroidSwitchNativeComponent.js b/src/private/specs_DEPRECATED/components/AndroidSwitchNativeComponent.js
index c91e734..1a51ca2 100644
--- a/src/private/specs_DEPRECATED/components/AndroidSwitchNativeComponent.js
+++ b/src/private/specs_DEPRECATED/components/AndroidSwitchNativeComponent.js
@@ -21,12 +21,12 @@ import codegenNativeCommands from '../../../../Libraries/Utilities/codegenNative
 import codegenNativeComponent from '../../../../Libraries/Utilities/codegenNativeComponent';
 import * as React from 'react';
 
-type AndroidSwitchChangeEvent = Readonly<{
+type AndroidSwitchChangeEvent = $ReadOnly<{
   value: boolean,
   target: Int32,
 }>;
 
-type AndroidSwitchNativeProps = Readonly<{
+type AndroidSwitchNativeProps = $ReadOnly<{
   ...ViewProps,
 
   // Props
diff --git a/src/private/specs_DEPRECATED/components/DebuggingOverlayNativeComponent.js b/src/private/specs_DEPRECATED/components/DebuggingOverlayNativeComponent.js
index 178951b..a7dff66 100644
--- a/src/private/specs_DEPRECATED/components/DebuggingOverlayNativeComponent.js
+++ b/src/private/specs_DEPRECATED/components/DebuggingOverlayNativeComponent.js
@@ -38,11 +38,11 @@ export type ElementRectangle = {
 interface NativeCommands {
   +highlightTraceUpdates: (
     viewRef: React.ElementRef<DebuggingOverlayNativeComponentType>,
-    updates: ReadonlyArray<TraceUpdate>,
+    updates: $ReadOnlyArray<TraceUpdate>,
   ) => void;
   +highlightElements: (
     viewRef: React.ElementRef<DebuggingOverlayNativeComponentType>,
-    elements: ReadonlyArray<ElementRectangle>,
+    elements: $ReadOnlyArray<ElementRectangle>,
   ) => void;
   +clearElementsHighlights: (
     viewRef: React.ElementRef<DebuggingOverlayNativeComponentType>,
diff --git a/src/private/specs_DEPRECATED/components/RCTModalHostViewNativeComponent.js b/src/private/specs_DEPRECATED/components/RCTModalHostViewNativeComponent.js
index a27f5b3..b5aa18a 100644
--- a/src/private/specs_DEPRECATED/components/RCTModalHostViewNativeComponent.js
+++ b/src/private/specs_DEPRECATED/components/RCTModalHostViewNativeComponent.js
@@ -18,11 +18,11 @@ import type {HostComponent} from '../../types/HostComponent';
 
 import codegenNativeComponent from '../../../../Libraries/Utilities/codegenNativeComponent';
 
-type OrientationChangeEvent = Readonly<{
+type OrientationChangeEvent = $ReadOnly<{
   orientation: 'portrait' | 'landscape',
 }>;
 
-type RCTModalHostViewNativeProps = Readonly<{
+type RCTModalHostViewNativeProps = $ReadOnly<{
   ...ViewProps,
 
   /**
@@ -124,7 +124,7 @@ type RCTModalHostViewNativeProps = Readonly<{
    * See https://reactnative.dev/docs/modal#supportedorientations
    */
   supportedOrientations?: WithDefault<
-    ReadonlyArray<
+    $ReadOnlyArray<
       | 'portrait'
       | 'portrait-upside-down'
       | 'landscape'
diff --git a/src/private/specs_DEPRECATED/components/SwitchNativeComponent.js b/src/private/specs_DEPRECATED/components/SwitchNativeComponent.js
index 3d6feb9..1f9f75a 100644
--- a/src/private/specs_DEPRECATED/components/SwitchNativeComponent.js
+++ b/src/private/specs_DEPRECATED/components/SwitchNativeComponent.js
@@ -21,12 +21,12 @@ import codegenNativeCommands from '../../../../Libraries/Utilities/codegenNative
 import codegenNativeComponent from '../../../../Libraries/Utilities/codegenNativeComponent';
 import * as React from 'react';
 
-type NativeSwitchChangeEvent = Readonly<{
+type NativeSwitchChangeEvent = $ReadOnly<{
   value: boolean,
   target: Int32,
 }>;
 
-type SwitchNativeProps = Readonly<{
+type SwitchNativeProps = $ReadOnly<{
   ...ViewProps,
 
   // Props
diff --git a/src/private/webapis/dom/nodes/specs/NativeDOM.js b/src/private/webapis/dom/nodes/specs/NativeDOM.js
index 28f9e44..63f68ce 100644
--- a/src/private/webapis/dom/nodes/specs/NativeDOM.js
+++ b/src/private/webapis/dom/nodes/specs/NativeDOM.js
@@ -52,25 +52,25 @@ export interface Spec extends TurboModule {
    */
 
   +compareDocumentPosition: (
-    nativeNodeReference: unknown /* NativeNodeReference */,
-    otherNativeNodeReference: unknown /* NativeNodeReference */,
+    nativeNodeReference: mixed /* NativeNodeReference */,
+    otherNativeNodeReference: mixed /* NativeNodeReference */,
   ) => number;
 
   +getChildNodes: (
-    nativeNodeReference: unknown /* NativeNodeReference */,
-  ) => ReadonlyArray<unknown> /* $ReadOnlyArray<InstanceHandle> */;
+    nativeNodeReference: mixed /* NativeNodeReference */,
+  ) => $ReadOnlyArray<mixed> /* $ReadOnlyArray<InstanceHandle> */;
 
   +getElementById?: (
-    nativeNodeReference: unknown /* NativeNodeReference */,
+    nativeNodeReference: mixed /* NativeNodeReference */,
     id: string,
-  ) => unknown /* ?InstanceHandle */;
+  ) => mixed /* ?InstanceHandle */;
 
   +getParentNode: (
-    nativeNodeReference: unknown /* NativeNodeReference */,
-  ) => unknown /* ?InstanceHandle */;
+    nativeNodeReference: mixed /* NativeNodeReference */,
+  ) => mixed /* ?InstanceHandle */;
 
   +isConnected: (
-    nativeNodeReference: unknown /* NativeNodeReference */,
+    nativeNodeReference: mixed /* NativeNodeReference */,
   ) => boolean;
 
   /*
@@ -78,46 +78,46 @@ export interface Spec extends TurboModule {
    */
 
   +getBorderWidth: (
-    nativeElementReference: unknown /* NativeElementReference */,
-  ) => ReadonlyArray<number> /* [topWidth: number, rightWidth: number, bottomWidth: number, leftWidth: number] */;
+    nativeElementReference: mixed /* NativeElementReference */,
+  ) => $ReadOnlyArray<number> /* [topWidth: number, rightWidth: number, bottomWidth: number, leftWidth: number] */;
 
   +getBoundingClientRect: (
-    nativeElementReference: unknown /* NativeElementReference */,
+    nativeElementReference: mixed /* NativeElementReference */,
     includeTransform: boolean,
-  ) => ReadonlyArray<number> /* [x: number, y: number, width: number, height: number] */;
+  ) => $ReadOnlyArray<number> /* [x: number, y: number, width: number, height: number] */;
 
   +getInnerSize: (
-    nativeElementReference: unknown /* NativeElementReference */,
-  ) => ReadonlyArray<number> /* [width: number, height: number] */;
+    nativeElementReference: mixed /* NativeElementReference */,
+  ) => $ReadOnlyArray<number> /* [width: number, height: number] */;
 
   +getScrollPosition: (
-    nativeElementReference: unknown /* NativeElementReference */,
-  ) => ReadonlyArray<number> /* [scrollLeft: number, scrollTop: number] */;
+    nativeElementReference: mixed /* NativeElementReference */,
+  ) => $ReadOnlyArray<number> /* [scrollLeft: number, scrollTop: number] */;
 
   +getScrollSize: (
-    nativeElementReference: unknown /* NativeElementReference */,
-  ) => ReadonlyArray<number> /* [scrollWidth: number, scrollHeight: number] */;
+    nativeElementReference: mixed /* NativeElementReference */,
+  ) => $ReadOnlyArray<number> /* [scrollWidth: number, scrollHeight: number] */;
 
   +getTagName: (
-    nativeElementReference: unknown /* NativeElementReference */,
+    nativeElementReference: mixed /* NativeElementReference */,
   ) => string;
 
   +getTextContent: (
-    nativeElementReference: unknown /* NativeElementReference */,
+    nativeElementReference: mixed /* NativeElementReference */,
   ) => string;
 
   +hasPointerCapture: (
-    nativeElementReference: unknown /* NativeElementReference */,
+    nativeElementReference: mixed /* NativeElementReference */,
     pointerId: number,
   ) => boolean;
 
   +releasePointerCapture: (
-    nativeElementReference: unknown /* NativeElementReference */,
+    nativeElementReference: mixed /* NativeElementReference */,
     pointerId: number,
   ) => void;
 
   +setPointerCapture: (
-    nativeElementReference: unknown /* NativeElementReference */,
+    nativeElementReference: mixed /* NativeElementReference */,
     pointerId: number,
   ) => void;
 
@@ -126,8 +126,8 @@ export interface Spec extends TurboModule {
    */
 
   +getOffset: (
-    nativeElementReference: unknown /* NativeElementReference */,
-  ) => ReadonlyArray<unknown> /* [offsetParent: ?InstanceHandle, top: number, left: number] */;
+    nativeElementReference: mixed /* NativeElementReference */,
+  ) => $ReadOnlyArray<mixed> /* [offsetParent: ?InstanceHandle, top: number, left: number] */;
 
   /*
    * Special methods to handle the root node.
@@ -135,26 +135,26 @@ export interface Spec extends TurboModule {
 
   +linkRootNode?: (
     rootTag: number /* RootTag */,
-    instanceHandle: unknown /* InstanceHandle */,
-  ) => unknown /* ?NativeElementReference */;
+    instanceHandle: mixed /* InstanceHandle */,
+  ) => mixed /* ?NativeElementReference */;
 
   /**
    * Legacy layout APIs (for `ReactNativeElement`).
    */
 
   +measure: (
-    nativeElementReference: unknown,
+    nativeElementReference: mixed,
     callback: MeasureOnSuccessCallback,
   ) => void;
 
   +measureInWindow: (
-    nativeElementReference: unknown,
+    nativeElementReference: mixed,
     callback: MeasureInWindowOnSuccessCallback,
   ) => void;
 
   +measureLayout: (
-    nativeElementReference: unknown,
-    relativeNode: unknown,
+    nativeElementReference: mixed,
+    relativeNode: mixed,
     onFail: () => void,
     onSuccess: MeasureLayoutOnSuccessCallback,
   ) => void;
@@ -164,8 +164,8 @@ export interface Spec extends TurboModule {
    */
 
   +setNativeProps: (
-    nativeElementReference: unknown,
-    updatePayload: unknown,
+    nativeElementReference: mixed,
+    updatePayload: mixed,
   ) => void;
 }
 
@@ -199,7 +199,7 @@ export interface RefinedSpec {
    */
   +getChildNodes: (
     nativeNodeReference: NativeNodeReference,
-  ) => ReadonlyArray<InstanceHandle>;
+  ) => $ReadOnlyArray<InstanceHandle>;
 
   /**
    * This is a React Native implementation of `Document.prototype.getElementById`
diff --git a/src/private/webapis/idlecallbacks/specs/NativeIdleCallbacks.js b/src/private/webapis/idlecallbacks/specs/NativeIdleCallbacks.js
index 1b5f835..4241e5e 100644
--- a/src/private/webapis/idlecallbacks/specs/NativeIdleCallbacks.js
+++ b/src/private/webapis/idlecallbacks/specs/NativeIdleCallbacks.js
@@ -12,7 +12,7 @@ import type {TurboModule} from '../../../../../Libraries/TurboModule/RCTExport';
 
 import * as TurboModuleRegistry from '../../../../../Libraries/TurboModule/TurboModuleRegistry';
 
-export opaque type IdleCallbackID = unknown;
+export opaque type IdleCallbackID = mixed;
 
 export type RequestIdleCallbackOptions = {
   timeout?: number,
@@ -20,12 +20,12 @@ export type RequestIdleCallbackOptions = {
 
 export type IdleDeadline = {
   didTimeout: boolean,
-  timeRemaining: () => unknown,
+  timeRemaining: () => mixed,
 };
 
 export interface Spec extends TurboModule {
   +requestIdleCallback: (
-    callback: (idleDeadline: IdleDeadline) => unknown,
+    callback: (idleDeadline: IdleDeadline) => mixed,
     options?: RequestIdleCallbackOptions,
   ) => IdleCallbackID;
   +cancelIdleCallback: (handle: IdleCallbackID) => void;
diff --git a/src/private/webapis/intersectionobserver/specs/NativeIntersectionObserver.js b/src/private/webapis/intersectionobserver/specs/NativeIntersectionObserver.js
index cbe579b..fed5c09 100644
--- a/src/private/webapis/intersectionobserver/specs/NativeIntersectionObserver.js
+++ b/src/private/webapis/intersectionobserver/specs/NativeIntersectionObserver.js
@@ -14,25 +14,25 @@ import * as TurboModuleRegistry from '../../../../../Libraries/TurboModule/Turbo
 
 export type NativeIntersectionObserverEntry = {
   intersectionObserverId: number,
-  targetInstanceHandle: unknown,
-  targetRect: ReadonlyArray<number>, // It's actually a tuple with x, y, width and height
-  rootRect: ReadonlyArray<number>, // It's actually a tuple with x, y, width and height
+  targetInstanceHandle: mixed,
+  targetRect: $ReadOnlyArray<number>, // It's actually a tuple with x, y, width and height
+  rootRect: $ReadOnlyArray<number>, // It's actually a tuple with x, y, width and height
   // TODO(T209328432) - Remove optionality of intersectionRect when native changes are released
-  intersectionRect: ?ReadonlyArray<number>, // It's actually a tuple with x, y, width and height
+  intersectionRect: ?$ReadOnlyArray<number>, // It's actually a tuple with x, y, width and height
   isIntersectingAboveThresholds: boolean,
   time: number,
 };
 
 export type NativeIntersectionObserverObserveOptions = {
   intersectionObserverId: number,
-  rootShadowNode?: ?unknown,
-  targetShadowNode: unknown,
-  thresholds: ReadonlyArray<number>,
-  rootThresholds?: ?ReadonlyArray<number>,
+  rootShadowNode?: ?mixed,
+  targetShadowNode: mixed,
+  thresholds: $ReadOnlyArray<number>,
+  rootThresholds?: ?$ReadOnlyArray<number>,
   rootMargin?: ?string,
 };
 
-export opaque type NativeIntersectionObserverToken = unknown;
+export opaque type NativeIntersectionObserverToken = mixed;
 
 export interface Spec extends TurboModule {
   +observeV2?: (
@@ -44,7 +44,7 @@ export interface Spec extends TurboModule {
   ) => void;
   +connect: (notifyIntersectionObserversCallback: () => void) => void;
   +disconnect: () => void;
-  +takeRecords: () => ReadonlyArray<NativeIntersectionObserverEntry>;
+  +takeRecords: () => $ReadOnlyArray<NativeIntersectionObserverEntry>;
 }
 
 export default (TurboModuleRegistry.get<Spec>(
diff --git a/src/private/webapis/microtasks/specs/NativeMicrotasks.js b/src/private/webapis/microtasks/specs/NativeMicrotasks.js
index c6be40a..4f07158 100644
--- a/src/private/webapis/microtasks/specs/NativeMicrotasks.js
+++ b/src/private/webapis/microtasks/specs/NativeMicrotasks.js
@@ -13,7 +13,7 @@ import type {TurboModule} from '../../../../../Libraries/TurboModule/RCTExport';
 import * as TurboModuleRegistry from '../../../../../Libraries/TurboModule/TurboModuleRegistry';
 
 export interface Spec extends TurboModule {
-  +queueMicrotask: (callback: () => unknown) => void;
+  +queueMicrotask: (callback: () => mixed) => void;
 }
 
 export default (TurboModuleRegistry.getEnforcing<Spec>(
diff --git a/src/private/webapis/mutationobserver/specs/NativeMutationObserver.js b/src/private/webapis/mutationobserver/specs/NativeMutationObserver.js
index ade4886..c4fbcb2 100644
--- a/src/private/webapis/mutationobserver/specs/NativeMutationObserver.js
+++ b/src/private/webapis/mutationobserver/specs/NativeMutationObserver.js
@@ -15,16 +15,16 @@ import * as TurboModuleRegistry from '../../../../../Libraries/TurboModule/Turbo
 export type MutationObserverId = number;
 
 // These types are not supported by the codegen.
-type ShadowNode = unknown;
-type InstanceHandle = unknown;
-type ReactNativeElement = unknown;
-type ReadOnlyNode = unknown;
+type ShadowNode = mixed;
+type InstanceHandle = mixed;
+type ReactNativeElement = mixed;
+type ReadOnlyNode = mixed;
 
 export type NativeMutationRecord = {
   mutationObserverId: MutationObserverId,
   target: ReactNativeElement,
-  addedNodes: ReadonlyArray<ReadOnlyNode>,
-  removedNodes: ReadonlyArray<ReadOnlyNode>,
+  addedNodes: $ReadOnlyArray<ReadOnlyNode>,
+  removedNodes: $ReadOnlyArray<ReadOnlyNode>,
   ...
 };
 
@@ -47,7 +47,7 @@ export interface Spec extends TurboModule {
     ) => ReadOnlyNode,
   ) => void;
   +disconnect: () => void;
-  +takeRecords: () => ReadonlyArray<NativeMutationRecord>;
+  +takeRecords: () => $ReadOnlyArray<NativeMutationRecord>;
 }
 
 export default (TurboModuleRegistry.get<Spec>(
diff --git a/src/private/webapis/performance/specs/NativePerformance.js b/src/private/webapis/performance/specs/NativePerformance.js
index 47025dd..af194df 100644
--- a/src/private/webapis/performance/specs/NativePerformance.js
+++ b/src/private/webapis/performance/specs/NativePerformance.js
@@ -42,14 +42,14 @@ export type RawPerformanceEntry = {
   decodedBodySize?: number,
 };
 
-export opaque type OpaqueNativeObserverHandle = unknown;
+export opaque type OpaqueNativeObserverHandle = mixed;
 
 export type NativeBatchedObserverCallback = () => void;
 export type NativePerformanceMarkResult = number;
-export type NativePerformanceMeasureResult = ReadonlyArray<number>; // [startTime, duration]
+export type NativePerformanceMeasureResult = $ReadOnlyArray<number>; // [startTime, duration]
 
 export type PerformanceObserverInit = {
-  entryTypes?: ReadonlyArray<number>,
+  entryTypes?: $ReadOnlyArray<number>,
   type?: number,
   buffered?: boolean,
   durationThreshold?: number,
@@ -59,25 +59,25 @@ export interface Spec extends TurboModule {
   +now: () => number;
   +timeOrigin?: () => number;
 
-  +reportMark: (name: string, startTime: number, entry: unknown) => void;
+  +reportMark: (name: string, startTime: number, entry: mixed) => void;
   +reportMeasure: (
     name: string,
     startTime: number,
     duration: number,
-    entry: unknown,
+    entry: mixed,
   ) => void;
   +getMarkTime: (name: string) => ?number;
   +clearMarks: (entryName?: string) => void;
   +clearMeasures: (entryName?: string) => void;
-  +getEntries: () => ReadonlyArray<RawPerformanceEntry>;
+  +getEntries: () => $ReadOnlyArray<RawPerformanceEntry>;
   +getEntriesByName: (
     entryName: string,
     entryType?: ?RawPerformanceEntryType,
-  ) => ReadonlyArray<RawPerformanceEntry>;
+  ) => $ReadOnlyArray<RawPerformanceEntry>;
   +getEntriesByType: (
     entryType: RawPerformanceEntryType,
-  ) => ReadonlyArray<RawPerformanceEntry>;
-  +getEventCounts: () => ReadonlyArray<[string, number]>;
+  ) => $ReadOnlyArray<RawPerformanceEntry>;
+  +getEventCounts: () => $ReadOnlyArray<[string, number]>;
   +getSimpleMemoryInfo: () => NativeMemoryInfo;
   +getReactNativeStartupTiming: () => ReactNativeStartupTiming;
 
@@ -94,9 +94,9 @@ export interface Spec extends TurboModule {
   +takeRecords: (
     observer: OpaqueNativeObserverHandle,
     sort: boolean,
-  ) => ReadonlyArray<RawPerformanceEntry>;
+  ) => $ReadOnlyArray<RawPerformanceEntry>;
 
-  +getSupportedPerformanceEntryTypes: () => ReadonlyArray<RawPerformanceEntryType>;
+  +getSupportedPerformanceEntryTypes: () => $ReadOnlyArray<RawPerformanceEntryType>;
 
   +clearEventCountsForTesting: () => void;
 }
