diff --git a/src/fabric/gamma/stack/StackHeaderConfigAndroidNativeComponent.ts b/src/fabric/gamma/stack/StackHeaderConfigAndroidNativeComponent.ts
index 945a75bebe089b8b6798e659031fbd91060d52eb..549fca693645bd3e61b11dfbff22b21f9e2e5f9f 100644
--- a/src/fabric/gamma/stack/StackHeaderConfigAndroidNativeComponent.ts
+++ b/src/fabric/gamma/stack/StackHeaderConfigAndroidNativeComponent.ts
@@ -81,7 +81,7 @@ export type StackHeaderToolbarMenuItemOptionsAndroid = Partial<
 
 export interface NativeCommands {
   setToolbarMenuItemOptions: (
-    viewRef: React.ComponentRef<ComponentType>,
+    viewRef: React.ElementRef<ComponentType>,
     id: string,
     // We use the array here only due to codegen limitation. We're using only
     // the first index of the array.
