diff --git a/node_modules/react-native/src/private/webapis/idlecallbacks/specs/NativeIdleCallbacks.js b/node_modules/react-native/src/private/webapis/idlecallbacks/specs/NativeIdleCallbacks.js
index 4241e5e..3939434 100644
--- a/node_modules/react-native/src/private/webapis/idlecallbacks/specs/NativeIdleCallbacks.js
+++ b/node_modules/react-native/src/private/webapis/idlecallbacks/specs/NativeIdleCallbacks.js
@@ -27,8 +27,8 @@ export interface Spec extends TurboModule {
   +requestIdleCallback: (
     callback: (idleDeadline: IdleDeadline) => mixed,
     options?: RequestIdleCallbackOptions,
-  ) => IdleCallbackID;
-  +cancelIdleCallback: (handle: IdleCallbackID) => void;
+  ) => mixed;
+  +cancelIdleCallback: (handle: mixed) => void;
 }

 export default (TurboModuleRegistry.getEnforcing<Spec>(
diff --git a/node_modules/react-native/src/private/webapis/intersectionobserver/specs/NativeIntersectionObserver.js b/node_modules/react-native/src/private/webapis/intersectionobserver/specs/NativeIntersectionObserver.js
index f5e3609..9c207d2 100644
--- a/node_modules/react-native/src/private/webapis/intersectionobserver/specs/NativeIntersectionObserver.js
+++ b/node_modules/react-native/src/private/webapis/intersectionobserver/specs/NativeIntersectionObserver.js
@@ -36,10 +36,10 @@ export opaque type NativeIntersectionObserverToken = mixed;
 export interface Spec extends TurboModule {
   +observeV2?: (
     options: NativeIntersectionObserverObserveOptions,
-  ) => NativeIntersectionObserverToken;
+  ) => mixed;
   +unobserveV2?: (
     intersectionObserverId: number,
-    token: NativeIntersectionObserverToken,
+    token: mixed,
   ) => void;
   +connect: (notifyIntersectionObserversCallback: () => void) => void;
   +disconnect: () => void;
diff --git a/node_modules/react-native/src/private/webapis/performance/specs/NativePerformance.js b/node_modules/react-native/src/private/webapis/performance/specs/NativePerformance.js
index e191d0a..603ae59 100644
--- a/node_modules/react-native/src/private/webapis/performance/specs/NativePerformance.js
+++ b/node_modules/react-native/src/private/webapis/performance/specs/NativePerformance.js
@@ -80,16 +80,16 @@ export interface Spec extends TurboModule {

   +createObserver: (
     callback: NativeBatchedObserverCallback,
-  ) => OpaqueNativeObserverHandle;
-  +getDroppedEntriesCount: (observer: OpaqueNativeObserverHandle) => number;
+  ) => mixed;
+  +getDroppedEntriesCount: (observer: mixed) => number;

   +observe: (
-    observer: OpaqueNativeObserverHandle,
+    observer: mixed,
     options: PerformanceObserverInit,
   ) => void;
-  +disconnect: (observer: OpaqueNativeObserverHandle) => void;
+  +disconnect: (observer: mixed) => void;
   +takeRecords: (
-    observer: OpaqueNativeObserverHandle,
+    observer: mixed,
     sort: boolean,
   ) => $ReadOnlyArray<RawPerformanceEntry>;

