--- android/CMakeLists.txt
+++ android/CMakeLists.txt
@@ -102,7 +102,7 @@ target_include_directories(
 )

 # Import prebuilt SKIA libraries
-set (SKIA_LIBS_PATH "${CMAKE_CURRENT_SOURCE_DIR}/../libs/android/${ANDROID_ABI}")
+set (SKIA_LIBS_PATH "${CMAKE_CURRENT_SOURCE_DIR}/../../../../common/libs/${ANDROID_ABI}")

 add_library(skia STATIC IMPORTED)
 set_property(TARGET skia PROPERTY IMPORTED_LOCATION "${SKIA_LIBS_PATH}/libskia.a")
--- android/build.gradle
+++ android/build.gradle
@@ -56,7 +56,7 @@ static def findNodeModules(baseDir) {
   throw new GradleException("React-Native-Skia: Failed to find node_modules/ path!")
 }

-def nodeModules = findNodeModules(projectDir)
+def nodeModules = Paths.get(projectDir.getPath(), '../../../../../..', 'react-native-lab/react-native/packages').toString()
 logger.warn("react-native-skia: node_modules/ found at: ${nodeModules}")

 def sourceBuild = false
@@ -68,9 +68,9 @@ if (isNewArchitectureEnabled()) {

 if (rootProject.ext.has('reactNativeAndroidRoot')) {
   defaultDir = rootProject.ext.get('reactNativeAndroidRoot')
-} else if (findProject(':ReactAndroid') != null) {
+} else if (findProject(':packages:react-native:ReactAndroid') != null) {
     sourceBuild = true
-    defaultDir = project(':ReactAndroid').projectDir
+    defaultDir = project(':packages:react-native:ReactAndroid').projectDir
 } else {
   defaultDir = file("$nodeModules/react-native")
 }
